diff --git a/cmake/external/eigen.cmake b/cmake/external/eigen.cmake
index cfada544d4..a690e97d74 100644
--- a/cmake/external/eigen.cmake
+++ b/cmake/external/eigen.cmake
@@ -42,12 +42,12 @@ endif()
 file(TO_NATIVE_PATH "${PADDLE_SOURCE_DIR}/patches/eigen/TensorRandom.h.patch"
      tensor_random_header)
 # See: [Why calling some `git` commands before `patch`?]
-set(EIGEN_PATCH_COMMAND git checkout -- . && git checkout ${EIGEN_TAG} && git
-                        apply ${tensor_random_header})
+# set(EIGEN_PATCH_COMMAND git checkout -- . && git checkout ${EIGEN_TAG} && git
+#                         apply ${tensor_random_header})
 if(CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang")
   file(TO_NATIVE_PATH ${PADDLE_SOURCE_DIR}/patches/eigen/Complex.h.patch
        complex_header)
-  set(EIGEN_PATCH_COMMAND ${EIGEN_PATCH_COMMAND} && git apply ${complex_header})
+  # set(EIGEN_PATCH_COMMAND ${EIGEN_PATCH_COMMAND} && git apply ${complex_header})
 endif()
 
 set(EIGEN_INCLUDE_DIR ${SOURCE_DIR})
diff --git a/paddle/fluid/platform/profiler/cupti_data_process.cc b/paddle/fluid/platform/profiler/cupti_data_process.cc
index bff0f2bf70..9376b5781f 100644
--- a/paddle/fluid/platform/profiler/cupti_data_process.cc
+++ b/paddle/fluid/platform/profiler/cupti_data_process.cc
@@ -16,7 +16,7 @@
 
 #include <cstdio>
 
-#include "paddle/fluid/platform/enforce.h"
+// #include "paddle/fluid/platform/enforce.h"
 #include "paddle/phi/core/os_info.h"
 #include "paddle/phi/core/platform/device/gpu/gpu_info.h"
 #include "paddle/phi/core/platform/profiler/utils.h"
diff --git a/paddle/phi/backends/dynload/cudnn.h b/paddle/phi/backends/dynload/cudnn.h
index 7a5450c349..95de89ced2 100644
--- a/paddle/phi/backends/dynload/cudnn.h
+++ b/paddle/phi/backends/dynload/cudnn.h
@@ -1,3 +1,4 @@
+// 2024 - Modified by MetaX Integrated Circuits (Shanghai) Co., Ltd. All Rights Reserved.   
 /* Copyright (c) 2016 PaddlePaddle Authors. All Rights Reserved.
 
 Licensed under the Apache License, Version 2.0 (the "License");
@@ -15,7 +16,6 @@ limitations under the License. */
 #pragma once
 #ifdef PADDLE_WITH_CUDA
 #include <cudnn.h>
-
 #include <mutex>  // NOLINT
 
 #include "paddle/phi/backends/dynload/dynamic_loader.h"
@@ -24,11 +24,11 @@ limitations under the License. */
 namespace phi {
 namespace dynload {
 
-TEST_API extern std::once_flag cudnn_dso_flag;
-TEST_API extern void* cudnn_dso_handle;
+extern std::once_flag cudnn_dso_flag;
+extern void* cudnn_dso_handle;
 extern bool HasCUDNN();
 
-TEST_API extern void EnforceCUDNNLoaded(const char* fn_name);
+extern void EnforceCUDNNLoaded(const char* fn_name);
 #define DECLARE_DYNAMIC_LOAD_CUDNN_WRAP(__name)                      \
   struct DynLoad__##__name {                                         \
     template <typename... Args>                                      \
@@ -38,7 +38,9 @@ TEST_API extern void EnforceCUDNNLoaded(const char* fn_name);
         cudnn_dso_handle = phi::dynload::GetCUDNNDsoHandle();        \
       });                                                            \
       EnforceCUDNNLoaded(#__name);                                   \
-      static void* p_##__name = dlsym(cudnn_dso_handle, #__name);    \
+      std::string replaced_name = #__name;                                  \
+      replaced_name =  replaced_name.replace(0,2,"mc");          \
+      static void* p_##__name = dlsym(cudnn_dso_handle, replaced_name.c_str());    \
       return reinterpret_cast<cudnn_func>(p_##__name)(args...);      \
     }                                                                \
   };                                                                 \
@@ -49,7 +51,6 @@ TEST_API extern void EnforceCUDNNLoaded(const char* fn_name);
  * different cudnn version has different interfaces
  **/
 #define CUDNN_DNN_ROUTINE_EACH(__macro)                    \
-  __macro(cudnnSetCallback);                               \
   __macro(cudnnSetTensor4dDescriptor);                     \
   __macro(cudnnSetTensor4dDescriptorEx);                   \
   __macro(cudnnSetTensorNdDescriptor);                     \
@@ -104,6 +105,13 @@ TEST_API extern void EnforceCUDNNLoaded(const char* fn_name);
   __macro(cudnnSetDropoutDescriptor);                      \
   __macro(cudnnRestoreDropoutDescriptor);                  \
   __macro(cudnnCreateRNNDescriptor);                       \
+  __macro(cudnnGetRNNParamsSize);                          \
+  __macro(cudnnGetRNNWorkspaceSize);                       \
+  __macro(cudnnGetRNNTrainingReserveSize);                 \
+  __macro(cudnnRNNForwardTraining);                        \
+  __macro(cudnnRNNBackwardData);                           \
+  __macro(cudnnRNNBackwardWeights);                        \
+  __macro(cudnnRNNForwardInference);                       \
   __macro(cudnnDestroyDropoutDescriptor);                  \
   __macro(cudnnDestroyRNNDescriptor);                      \
   __macro(cudnnSetTensorNdDescriptorEx);                   \
@@ -118,7 +126,8 @@ TEST_API extern void EnforceCUDNNLoaded(const char* fn_name);
   __macro(cudnnCreateActivationDescriptor);                \
   __macro(cudnnSetActivationDescriptor);                   \
   __macro(cudnnGetActivationDescriptor);                   \
-  __macro(cudnnDestroyActivationDescriptor);
+  __macro(cudnnDestroyActivationDescriptor);               \
+  __macro(cudnnSetRNNDescriptor_v6);
 CUDNN_DNN_ROUTINE_EACH(DECLARE_DYNAMIC_LOAD_CUDNN_WRAP)
 
 #if CUDNN_VERSION >= 7000 && CUDNN_VERSION < 8000
@@ -152,7 +161,12 @@ CUDNN_DNN_ROUTINE_EACH_R7(DECLARE_DYNAMIC_LOAD_CUDNN_WRAP)
 #define CUDNN_DNN_ROUTINE_EACH_AFTER_TWO_R7(__macro) \
   __macro(cudnnCreateRNNDataDescriptor);             \
   __macro(cudnnDestroyRNNDataDescriptor);            \
-  __macro(cudnnSetRNNDataDescriptor);
+  __macro(cudnnSetRNNDataDescriptor);                \
+  __macro(cudnnSetRNNPaddingMode);                   \
+  __macro(cudnnRNNForwardTrainingEx);                \
+  __macro(cudnnRNNBackwardDataEx);                   \
+  __macro(cudnnRNNBackwardWeightsEx);                \
+  __macro(cudnnRNNForwardInferenceEx);
 CUDNN_DNN_ROUTINE_EACH_AFTER_TWO_R7(DECLARE_DYNAMIC_LOAD_CUDNN_WRAP)
 #endif
 
@@ -195,40 +209,6 @@ CUDNN_DNN_ROUTINE_EACH_R8(DECLARE_DYNAMIC_LOAD_CUDNN_WRAP)
 CUDNN_DNN_ROUTINE_EACH_FRONTEND(DECLARE_DYNAMIC_LOAD_CUDNN_WRAP)
 #endif
 
-#if CUDNN_VERSION < 90000
-#define CUDNN_DNN_ROUTINE_EACH_REMOVED_IN_E9(__macro) \
-  __macro(cudnnGetRNNParamsSize);                     \
-  __macro(cudnnGetRNNWorkspaceSize);                  \
-  __macro(cudnnGetRNNTrainingReserveSize);            \
-  __macro(cudnnSetRNNDescriptor_v6);                  \
-  __macro(cudnnRNNForwardInference);                  \
-  __macro(cudnnRNNForwardTraining);                   \
-  __macro(cudnnRNNBackwardData);                      \
-  __macro(cudnnRNNBackwardWeights);
-CUDNN_DNN_ROUTINE_EACH_REMOVED_IN_E9(DECLARE_DYNAMIC_LOAD_CUDNN_WRAP)
-#endif
-
-#if CUDNN_VERSION < 90000 && CUDNN_VERSION >= 7201
-#define CUDNN_DNN_ROUTINE_EACH_AFTER_TWO_R7_REMOVED_IN_E9(__macro) \
-  __macro(cudnnSetRNNPaddingMode);                                 \
-  __macro(cudnnRNNForwardInferenceEx);                             \
-  __macro(cudnnRNNForwardTrainingEx);                              \
-  __macro(cudnnRNNBackwardDataEx);                                 \
-  __macro(cudnnRNNBackwardWeightsEx);
-CUDNN_DNN_ROUTINE_EACH_AFTER_TWO_R7_REMOVED_IN_E9(
-    DECLARE_DYNAMIC_LOAD_CUDNN_WRAP)
-#endif
-
-#if CUDNN_VERSION >= 90000
-#define CUDNN_DNN_ROUTINE_EACH_R9(__macro) \
-  __macro(cudnnGetLastErrorString);        \
-  __macro(cudnnGetRNNWeightSpaceSize);     \
-  __macro(cudnnGetRNNTempSpaceSizes);      \
-  __macro(cudnnRNNForward);                \
-  __macro(cudnnRNNBackwardData_v8);        \
-  __macro(cudnnRNNBackwardWeights_v8);
-CUDNN_DNN_ROUTINE_EACH_R9(DECLARE_DYNAMIC_LOAD_CUDNN_WRAP)
-#endif
 }  // namespace dynload
 }  // namespace phi
 
diff --git a/paddle/phi/backends/dynload/cupti.h b/paddle/phi/backends/dynload/cupti.h
index 59e92955c9..d2f8c2da15 100644
--- a/paddle/phi/backends/dynload/cupti.h
+++ b/paddle/phi/backends/dynload/cupti.h
@@ -24,8 +24,8 @@ limitations under the License. */
 #include "paddle/phi/backends/dynload/dynamic_loader.h"
 #include "paddle/phi/common/port.h"
 
-namespace phi {
-namespace dynload {
+// namespace phi {
+// namespace dynload {
 
 extern std::once_flag cupti_dso_flag;
 extern void *cupti_dso_handle;
@@ -71,7 +71,7 @@ extern void *cupti_dso_handle;
 CUPTI_ROUTINE_EACH(DECLARE_DYNAMIC_LOAD_CUPTI_WRAP);
 
 #undef DECLARE_DYNAMIC_LOAD_CUPTI_WRAP
-}  // namespace dynload
-}  // namespace phi
+// }  // namespace dynload
+// }  // namespace phi
 
-#endif  // PADDLE_WITH_CUPTI
+#endif  // PADDLE_WITH_CUPTI
\ No newline at end of file
diff --git a/paddle/phi/backends/dynload/cusolver.h b/paddle/phi/backends/dynload/cusolver.h
index 86651fc8f1..7c9b122a17 100644
--- a/paddle/phi/backends/dynload/cusolver.h
+++ b/paddle/phi/backends/dynload/cusolver.h
@@ -34,7 +34,9 @@ extern void *cusolver_dso_handle;
       std::call_once(cusolver_dso_flag, []() {                       \
         cusolver_dso_handle = phi::dynload::GetCusolverDsoHandle();  \
       });                                                            \
-      static void *p_##__name = dlsym(cusolver_dso_handle, #__name); \
+      std::string replaced_name = #__name;                                  \
+      replaced_name =  replaced_name.replace(0,2,"mc");          \
+      static void* p_##__name = dlsym(cusolver_dso_handle, replaced_name.c_str());    \
       return reinterpret_cast<cusolverFunc>(p_##__name)(args...);    \
     }                                                                \
   };                                                                 \
diff --git a/paddle/phi/backends/dynload/cusparse.h b/paddle/phi/backends/dynload/cusparse.h
index 8ec3cf2792..6f5460df00 100644
--- a/paddle/phi/backends/dynload/cusparse.h
+++ b/paddle/phi/backends/dynload/cusparse.h
@@ -34,7 +34,9 @@ extern void *cusparse_dso_handle;
       std::call_once(cusparse_dso_flag, []() {                       \
         cusparse_dso_handle = phi::dynload::GetCusparseDsoHandle();  \
       });                                                            \
-      static void *p_##__name = dlsym(cusparse_dso_handle, #__name); \
+      std::string replaced_name = #__name;                                  \
+      replaced_name =  replaced_name.replace(0,2,"mc");          \
+      static void* p_##__name = dlsym(cusparse_dso_handle, replaced_name.c_str());    \
       return reinterpret_cast<Func>(p_##__name)(args...);            \
     }                                                                \
   };                                                                 \
diff --git a/paddle/phi/backends/dynload/nvjpeg.h b/paddle/phi/backends/dynload/nvjpeg.h
index c5309e7e11..3328571380 100644
--- a/paddle/phi/backends/dynload/nvjpeg.h
+++ b/paddle/phi/backends/dynload/nvjpeg.h
@@ -31,7 +31,9 @@ extern void *nvjpeg_dso_handle;
       std::call_once(nvjpeg_dso_flag, []() {                       \
         nvjpeg_dso_handle = phi::dynload::GetNvjpegDsoHandle();    \
       });                                                          \
-      static void *p_##__name = dlsym(nvjpeg_dso_handle, #__name); \
+      std::string replaced_name = #__name;                                  \
+      replaced_name =  replaced_name.replace(0,2,"mc");          \
+      static void* p_##__name = dlsym(nvjpeg_dso_handle, replaced_name.c_str());    \
       return reinterpret_cast<nvjpegFunc>(p_##__name)(args...);    \
     }                                                              \
   };                                                               \
diff --git a/paddle/phi/backends/gpu/cuda/cuda_device_function.h b/paddle/phi/backends/gpu/cuda/cuda_device_function.h
index 4ff2e528a9..81421c8ca1 100644
--- a/paddle/phi/backends/gpu/cuda/cuda_device_function.h
+++ b/paddle/phi/backends/gpu/cuda/cuda_device_function.h
@@ -1,3 +1,4 @@
+// 2024 - Modified by MetaX Integrated Circuits (Shanghai) Co., Ltd. All Rights Reserved.   
 /* Copyright (c) 2022 PaddlePaddle Authors. All Rights Reserved.
 
 Licensed under the Apache License, Version 2.0 (the "License");
@@ -25,7 +26,7 @@ namespace phi {
 namespace backends {
 namespace gpu {
 
-#define FULL_WARP_MASK 0xFFFFFFFF
+#define FULL_WARP_MASK 0xFFFFFFFFFFFFFFFFULL
 #define CREATE_SHFL_MASK(mask, predicate) \
   mask = __ballot_sync(FULL_WARP_MASK, (predicate))
 
@@ -45,12 +46,12 @@ namespace gpu {
 
 template <typename T>
 __forceinline__ __device__ T
-CudaShuffleDownSync(unsigned mask, T val, int delta, int width = warpSize) {
+CudaShuffleDownSync(unsigned long long mask, T val, int delta, int width = warpSize) {
   return __shfl_down_sync(mask, val, static_cast<unsigned>(delta), width);
 }
 
 template <typename T>
-__forceinline__ __device__ T CudaShuffleXorSync(unsigned mask,
+__forceinline__ __device__ T CudaShuffleXorSync(unsigned long long mask,
                                                 T val,
                                                 int width = warpSize) {
   return __shfl_xor_sync(mask, val, width);
@@ -58,14 +59,14 @@ __forceinline__ __device__ T CudaShuffleXorSync(unsigned mask,
 
 template <>
 __forceinline__ __device__ phi::dtype::float16 CudaShuffleDownSync(
-    unsigned mask, phi::dtype::float16 val, int delta, int width) {
+    unsigned long long mask, phi::dtype::float16 val, int delta, int width) {
   return phi::dtype::float16(__shfl_down_sync(
       mask, val.to_half(), static_cast<unsigned>(delta), width));
 }
 
 template <>
 __forceinline__ __device__ phi::dtype::bfloat16 CudaShuffleDownSync(
-    unsigned mask, phi::dtype::bfloat16 val, int delta, int width) {
+    unsigned long long mask, phi::dtype::bfloat16 val, int delta, int width) {
 #if defined(PADDLE_CUDA_BF16)
   return phi::dtype::bfloat16(__shfl_down_sync(
       mask, val.to_nv_bfloat16(), static_cast<unsigned>(delta), width));
@@ -77,7 +78,7 @@ __forceinline__ __device__ phi::dtype::bfloat16 CudaShuffleDownSync(
 
 template <>
 __forceinline__ __device__ phi::dtype::complex<float> CudaShuffleDownSync(
-    unsigned mask, phi::dtype::complex<float> val, int delta, int width) {
+    unsigned long long mask, phi::dtype::complex<float> val, int delta, int width) {
   float real = static_cast<float>(__shfl_down_sync(
       mask, static_cast<float>(val.real), static_cast<unsigned>(delta), width));
   float imag = static_cast<float>(__shfl_down_sync(
@@ -87,7 +88,7 @@ __forceinline__ __device__ phi::dtype::complex<float> CudaShuffleDownSync(
 
 template <>
 __forceinline__ __device__ phi::dtype::complex<double> CudaShuffleDownSync(
-    unsigned mask, phi::dtype::complex<double> val, int delta, int width) {
+    unsigned long long mask, phi::dtype::complex<double> val, int delta, int width) {
   double real =
       static_cast<double>(__shfl_down_sync(mask,
                                            static_cast<double>(val.real),
@@ -103,13 +104,13 @@ __forceinline__ __device__ phi::dtype::complex<double> CudaShuffleDownSync(
 
 template <>
 __forceinline__ __device__ phi::dtype::float16 CudaShuffleXorSync(
-    unsigned mask, phi::dtype::float16 val, int width) {
+    unsigned long long mask, phi::dtype::float16 val, int width) {
   return phi::dtype::float16(__shfl_xor_sync(mask, val.to_half(), width));
 }
 
 template <>
 __forceinline__ __device__ phi::dtype::bfloat16 CudaShuffleXorSync(
-    unsigned mask, phi::dtype::bfloat16 val, int width) {
+    unsigned long long mask, phi::dtype::bfloat16 val, int width) {
 #if defined(PADDLE_CUDA_BF16)
   return phi::dtype::bfloat16(
       __shfl_xor_sync(mask, val.to_nv_bfloat16(), width));
@@ -121,7 +122,7 @@ __forceinline__ __device__ phi::dtype::bfloat16 CudaShuffleXorSync(
 
 template <>
 __forceinline__ __device__ phi::dtype::complex<float> CudaShuffleXorSync(
-    unsigned mask, phi::dtype::complex<float> val, int width) {
+    unsigned long long mask, phi::dtype::complex<float> val, int width) {
   float real = static_cast<float>(
       __shfl_xor_sync(mask, static_cast<float>(val.real), width));
   float imag = static_cast<float>(
@@ -131,7 +132,7 @@ __forceinline__ __device__ phi::dtype::complex<float> CudaShuffleXorSync(
 
 template <>
 __forceinline__ __device__ phi::dtype::complex<double> CudaShuffleXorSync(
-    unsigned mask, phi::dtype::complex<double> val, int width) {
+    unsigned long long mask, phi::dtype::complex<double> val, int width) {
   double real = static_cast<double>(
       __shfl_xor_sync(mask, static_cast<double>(val.real), width));
   double imag = static_cast<double>(
@@ -141,7 +142,7 @@ __forceinline__ __device__ phi::dtype::complex<double> CudaShuffleXorSync(
 
 template <typename T>
 __forceinline__ __device__ T
-CudaShuffleSync(unsigned mask, T val, int src_line, int width = 32) {
+CudaShuffleSync(unsigned long long mask, T val, int src_line, int width = 32) {
   return __shfl_sync(mask, val, src_line, width);
 }
 
@@ -160,7 +161,7 @@ __device__ T reduceSum(T val, int tid, int len) {
   // but most card's warp size is 32.
   const int warpSize = 32;
   __shared__ T shm[warpSize];
-  unsigned mask = 0u;
+  unsigned long long mask = 0ull;
   CREATE_SHFL_MASK(mask, tid < len);
 
   for (int offset = warpSize / 2; offset > 0; offset /= 2)
diff --git a/paddle/phi/core/enforce.h b/paddle/phi/core/enforce.h
index 95f1d58c64..667064f341 100644
--- a/paddle/phi/core/enforce.h
+++ b/paddle/phi/core/enforce.h
@@ -45,7 +45,9 @@ limitations under the License. */
 #endif
 
 #ifdef PADDLE_WITH_CUDA
-#include "paddle/phi/backends/dynload/cublas.h"
+// #include "paddle/phi/backends/dynload/../../../../../cublas.h"
+#include "../backends/metax_gpu/kernels/funcs/blas/cublas.h"
+// #include "paddle/phi/backends/dynload/cublas.h"
 #include "paddle/phi/backends/dynload/cudnn.h"
 #include "paddle/phi/backends/dynload/curand.h"
 #include "paddle/phi/backends/dynload/cusolver.h"
@@ -97,7 +99,7 @@ inline bool is_error(bool stat) { return !stat; }
 
 void ThrowWarnInternal(const std::string& message);
 
-#if defined(__CUDA_ARCH__)
+#if defined(__CUDACC__)
 // For cuda, the assertions can affect performance and it is therefore
 // recommended to disable them in production code
 // https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#assertion
@@ -109,7 +111,7 @@ void ThrowWarnInternal(const std::string& message);
              __LINE__,                                             \
              #_IS_NOT_ERROR,                                       \
              ##__VA_ARGS__);                                       \
-      asm("trap;");                                                \
+      __builtin_trap();                                             \
     }                                                              \
   } while (0)
 #elif defined(__HIPCC__)
@@ -757,4 +759,4 @@ inline void retry_sleep(unsigned millisecond) {
 
 }  // namespace enforce
 using namespace enforce;  // NOLINT
-}  // namespace phi
+}  // namespace phi
\ No newline at end of file
diff --git a/paddle/phi/core/platform/device/gpu/gpu_types.h b/paddle/phi/core/platform/device/gpu/gpu_types.h
index c646e487d0..325122175c 100644
--- a/paddle/phi/core/platform/device/gpu/gpu_types.h
+++ b/paddle/phi/core/platform/device/gpu/gpu_types.h
@@ -25,8 +25,9 @@
 #else
 #include <cuda_runtime.h>
 
-#include "paddle/phi/backends/dynload/cublas.h"
-#include "paddle/phi/backends/dynload/cublasLt.h"
+// #include "paddle/phi/backends/dynload/cublas.h"
+#include "kernels/funcs/blas/cublas.h"
+// #include "paddle/phi/backends/dynload/cublasLt.h"
 #include "paddle/phi/backends/dynload/cudnn.h"
 #endif
 
@@ -90,7 +91,7 @@ DECLARE_TYPE_FOR_GPU(gpuStreamCaptureMode,
 
 // TODO(Ming Huang): Since there is no blasLt handler,
 // use rocblas_handle for workaround.
-DECLARE_TYPE_FOR_GPU(blasLtHandle_t, cublasLtHandle_t, rocblas_handle);
+// DECLARE_TYPE_FOR_GPU(blasLtHandle_t, cublasLtHandle_t, rocblas_handle);
 
 #undef DECLARE_TYPE_FOR_GPU
 
diff --git a/paddle/phi/core/platform/device_context.h b/paddle/phi/core/platform/device_context.h
index d0526a99bd..f2db6354da 100644
--- a/paddle/phi/core/platform/device_context.h
+++ b/paddle/phi/core/platform/device_context.h
@@ -25,8 +25,8 @@ limitations under the License. */
 #include "paddle/phi/core/platform/device/gpu/gpu_types.h"
 #include "paddle/phi/core/platform/device_type.h"
 #ifdef PADDLE_WITH_CUDA
-#include "paddle/phi/backends/dynload/cublas.h"
-#include "paddle/phi/backends/dynload/cublasLt.h"
+#include "kernels/funcs/blas/cublas.h"
+#include "kernels/funcs/blas/cublasLt.h"
 #include "paddle/phi/backends/dynload/cudnn.h"
 #include "paddle/phi/backends/dynload/cusolver.h"
 #include "paddle/phi/backends/dynload/cusparse.h"
diff --git a/paddle/phi/kernels/funcs/fc_functor.cu b/paddle/phi/kernels/funcs/fc_functor.cu
index bdfd7313af..546bd07d5e 100644
--- a/paddle/phi/kernels/funcs/fc_functor.cu
+++ b/paddle/phi/kernels/funcs/fc_functor.cu
@@ -16,12 +16,12 @@ limitations under the License. */
 
 #include "paddle/phi/backends/all_context.h"
 #include "paddle/phi/kernels/funcs/aligned_vector.h"
-#include "paddle/phi/kernels/funcs/blas/blas.h"
+#include "kernels/funcs/blas/blas.h"
 #include "paddle/phi/kernels/funcs/fc_functor.h"
 
 #include "paddle/phi/backends/gpu/gpu_launch_config.h"
 #include "paddle/phi/core/dense_tensor.h"
-#include "paddle/phi/kernels/funcs/blas/blaslt_impl.cu.h"
+// #include "paddle/phi/kernels/funcs/blas/blaslt_impl.cu.h"
 #include "paddle/phi/kernels/funcs/quant_dequant.h"
 #include "paddle/phi/kernels/matmul_kernel.h"
 
diff --git a/paddle/phi/kernels/funcs/matrix_inverse.cu b/paddle/phi/kernels/funcs/matrix_inverse.cu
index 1a9a9cfb85..08ebe4b8af 100644
--- a/paddle/phi/kernels/funcs/matrix_inverse.cu
+++ b/paddle/phi/kernels/funcs/matrix_inverse.cu
@@ -15,11 +15,13 @@ limitations under the License. */
 #include "paddle/phi/kernels/funcs/matrix_inverse.h"
 
 #include "paddle/phi/common/memory_utils.h"
-#include "paddle/phi/kernels/funcs/blas/blas.h"
+#include "kernels/funcs/blas/blas.h"
 
 namespace phi {
 namespace funcs {
 
+
+
 template <typename Context, typename T>
 void MatrixInverseFunctor<Context, T>::operator()(const Context& dev_ctx,
                                                   const DenseTensor& a,
diff --git a/paddle/phi/kernels/funcs/matrix_solve.cu b/paddle/phi/kernels/funcs/matrix_solve.cu
index 558d363b39..05da04b517 100644
--- a/paddle/phi/kernels/funcs/matrix_solve.cu
+++ b/paddle/phi/kernels/funcs/matrix_solve.cu
@@ -16,7 +16,7 @@ limitations under the License. */
 #include "paddle/phi/backends/gpu/cuda/cudnn_workspace_helper.h"
 #include "paddle/phi/common/memory_utils.h"
 #include "paddle/phi/core/tensor_utils.h"
-#include "paddle/phi/kernels/funcs/blas/blas.h"
+#include "kernels/funcs/blas/blas.h"
 #include "paddle/phi/kernels/funcs/math_function.h"
 #include "paddle/phi/kernels/funcs/scatter.cu.h"
 
diff --git a/paddle/phi/kernels/funcs/top_k_function_cuda.h b/paddle/phi/kernels/funcs/top_k_function_cuda.h
index dc7935423c..84896c2214 100644
--- a/paddle/phi/kernels/funcs/top_k_function_cuda.h
+++ b/paddle/phi/kernels/funcs/top_k_function_cuda.h
@@ -32,11 +32,11 @@ limitations under the License. */
 #include "paddle/phi/kernels/funcs/eigen/eigen_function.h"
 #include "paddle/phi/kernels/primitive/functor_primitives.h"
 
-#define FINAL_MASK 0xffffffff
+#define FINAL_MASK 0xffffffffffffffffull
 #ifdef PADDLE_WITH_HIP
 #define WARP_SIZE 64
 #else
-#define WARP_SIZE 32
+#define WARP_SIZE 64
 #endif
 #define MAX_NUM_THREADS 1024
 
@@ -200,21 +200,56 @@ __device__ __forceinline__ void AddTo(Pair<T> topk[],
   for (int k = beam_size - 2; k >= 0; k--) {
     if (largest) {
       if (topk[k] < p) {
-        topk[k + 1] = topk[k];
+        topk[k + 1].v = topk[k].v;
+        topk[k + 1].id = topk[k].id;
       } else {
-        topk[k + 1] = p;
+        topk[k + 1].v = p.v;
+        topk[k + 1].id = p.id;
         return;
       }
     } else {
       if (topk[k] > p) {
-        topk[k + 1] = topk[k];
+        topk[k + 1].v = topk[k].v;
+        topk[k + 1].id = topk[k].id;
       } else {
-        topk[k + 1] = p;
+        topk[k + 1].v = p.v;
+        topk[k + 1].id = p.id;
         return;
       }
     }
   }
-  topk[0] = p;
+  topk[0].v = p.v;
+  topk[0].id = p.id;
+}
+
+template <typename T>
+__device__ __forceinline__ void AddTo(Pair<T> topk[],
+                                      const Pair<T>& p,
+                                      int beam_size,
+                                      const bool& largest, const int offset) {
+  for (int k = beam_size - 2; k >= 0; k--) {
+    if (largest) {
+      if (topk[k + offset] < p) {
+        topk[k + 1 + offset].v = topk[k + offset].v;
+        topk[k + 1 + offset].id = topk[k + offset].id;
+      } else {
+        topk[k + 1 + offset].v = p.v;
+        topk[k + 1 + offset].id = p.id;
+        return;
+      }
+    } else {
+      if (topk[k + offset] > p) {
+        topk[k + 1 + offset].v = topk[k + offset].v;
+        topk[k + 1 + offset].id = topk[k + offset].id;
+      } else {
+        topk[k + 1 + offset].v = p.v;
+        topk[k + 1 + offset].id = p.id;
+        return;
+      }
+    }
+  }
+  topk[0 + offset].v = p.v;
+  topk[0 + offset].id = p.id;
 }
 
 template <typename T, int BlockSize>
@@ -243,24 +278,24 @@ __device__ __forceinline__ void GetTopK(Pair<T> topk[],
 template <typename T, int BlockSize>
 __device__ __forceinline__ void GetTopK(Pair<T> topk[],
                                         const T* src,
-                                        int64_t idx,
-                                        int64_t dim,
+                                        int idx,
+                                        int dim,
                                         const Pair<T>& max,
                                         int beam_size,
-                                        const bool& largest) {
+                                        const bool& largest, const int offset) {
   while (idx < dim) {
     if (largest) {
-      if (topk[beam_size - 1] < src[idx]) {
+      if (topk[beam_size - 1 + offset] < src[idx]) {
         Pair<T> tmp(src[idx], idx);
         if (tmp < max) {
-          AddTo<T>(topk, tmp, beam_size, largest);
+          AddTo<T>(topk, tmp, beam_size, largest,offset);
         }
       }
     } else {
-      if (topk[beam_size - 1] > src[idx]) {
+      if (topk[beam_size - 1 + offset] > src[idx]) {
         Pair<T> tmp(src[idx], idx);
         if (tmp > max) {
-          AddTo<T>(topk, tmp, beam_size, largest);
+          AddTo<T>(topk, tmp, beam_size, largest,offset);
         }
       }
     }
@@ -287,7 +322,9 @@ __device__ __forceinline__ void ThreadGetTopK(Pair<T> topk[],
     } else {
       for (int k = 0; k < MaxLength; k++) {
         if (k < MaxLength - (*beam)) {
-          topk[k] = topk[k + *beam];
+          topk[k].v = topk[k + *beam].v;
+          topk[k].id = topk[k + *beam].id;
+
         } else {
           if (largest) {
             topk[k].set(-static_cast<T>(INFINITY), -1);
@@ -297,8 +334,10 @@ __device__ __forceinline__ void ThreadGetTopK(Pair<T> topk[],
         }
       }
       if (!(*is_empty)) {
+        const int offset =  MaxLength - *beam;
         GetTopK<T, BlockSize>(
-            topk + MaxLength - *beam, src, tid, dim, *max, length, largest);
+            topk, src, tid, dim, *max, length, largest, offset);
+            // topk + MaxLength - *beam, src, tid, dim, *max, length, largest);
       }
     }
 
@@ -359,6 +398,8 @@ __device__ __forceinline__ void BlockReduce(Pair<T> shared_max[],
       shared_max[wid] = input_now;
     }
     __syncthreads();
+    if(BlockSize > WARP_SIZE)
+    {
     if (largest) {
       input_now = (tid < BlockSize / WARP_SIZE)
                       ? shared_max[lane]
@@ -373,27 +414,32 @@ __device__ __forceinline__ void BlockReduce(Pair<T> shared_max[],
       if (lane == 0) shared_max[0] = input_now;
     }
     __syncthreads();
-
+    }
     if (tid == 0) {
       **topVal = input_now.v;
       **topIds = input_now.id;
       (*topVal)++;
       (*topIds)++;
     }
+    if (--(*k) == 0) break;
     int tid_max = shared_max[0].id % BlockSize;
     if (tid == tid_max) {
       (*beam)++;
       if (*beam < MaxLength) {
-        topk[0] = topk[*beam];
+        topk[0].v = topk[*beam].v;
+        topk[0].id = topk[*beam].id;
       }
     }
-    if (--(*k) == 0) break;
+    // if (--(*k) == 0) break;
+    unsigned long long mask = 0ull;
 
-    unsigned mask = 0u;
+    // unsigned mask = 0u;
     CREATE_SHFL_MASK(mask, true);
     if (tid_max / WARP_SIZE == wid) {
-      if (phi::backends::gpu::CudaShuffleSync(
-              mask, *beam, tid_max % WARP_SIZE, WARP_SIZE) == MaxLength)
+      // if (phi::backends::gpu::CudaShuffleSync(
+      //         mask, *beam, tid_max % WARP_SIZE, WARP_SIZE) == MaxLength)
+    if (phi::backends::gpu::CudaShuffleSync(
+                 mask, *beam, tid_max % WARP_SIZE, WARP_SIZE) == MaxLength)
         break;
     }
   }
@@ -482,16 +528,17 @@ struct Bitfield<unsigned int> {
                                                              int pos,
                                                              int len) {
     unsigned int ret;
-    asm("bfe.u32 %0, %1, %2, %3;" : "=r"(ret) : "r"(val), "r"(pos), "r"(len));
+  ret = (static_cast<unsigned int>(val) << (32 - pos - len)) >> (32 - len);
+
+
     return ret;
   }
 
   static __device__ __forceinline__ unsigned int SetBitfield(
       unsigned int val, unsigned int to_insert, int pos, int len) {
     unsigned int ret;
-    asm("bfi.b32 %0, %1, %2, %3, %4;"
-        : "=r"(ret)
-        : "r"(to_insert), "r"(val), "r"(pos), "r"(len));
+    
+    ret = (static_cast<unsigned int>(val) << (32 - pos - len)) >> (32 - len);
     return ret;
   }
 };
@@ -502,7 +549,9 @@ struct Bitfield<uint64_t> {
                                                          int pos,
                                                          int len) {
     uint64_t ret;
-    asm("bfe.u64 %0, %1, %2, %3;" : "=l"(ret) : "l"(val), "r"(pos), "r"(len));
+    
+
+    ret = (static_cast<uint64_t>(val) << (64 - pos - len)) >> (64 - len);
     return ret;
   }
 
@@ -511,9 +560,9 @@ struct Bitfield<uint64_t> {
                                                          int pos,
                                                          int len) {
     uint64_t ret;
-    asm("bfi.b64 %0, %1, %2, %3, %4;"
-        : "=l"(ret)
-        : "l"(to_insert), "l"(val), "r"(pos), "r"(len));
+    
+  ret = (static_cast<uint64_t>(val) << (64 - pos - len)) >> (64 - len);
+
     return ret;
   }
 };
@@ -631,14 +680,20 @@ struct RadixTypeConfig<phi::dtype::bfloat16> {
 /*---------------------------Helper Functions------------------*/
 __device__ __forceinline__ int GetLaneId() {
   int lane_id;
-  asm("mov.s32 %0, %%laneid;" : "=r"(lane_id));
-  return lane_id;
+  
+// // >>>> PTX2CPP Success <<<<
+// {
+// (lane_id)=(threadIdx.x&(warpSize-1));
+// }
+  return ::__lane_id();
+  // return lane_id;
 }
 
 __device__ __forceinline__ unsigned GetLaneMaskLe() {
   unsigned mask;
-  asm("mov.u32 %0, %%lanemask_le;" : "=r"(mask));
-  return mask;
+
+  return ((uint64_t(1) << ::__lane_id()) << 1) - 1;
+  // return mask;
 }
 
 template <typename T, bool KillDependency, class Function>
@@ -885,7 +940,8 @@ __global__ void GatherKthValue(const T* input,
 
   // 1. Find the k-th value
   T kth_value = static_cast<T>(0);
-  RadixSearch<T, RadixTypeConfig<T>::RadixType, IndexType, false>(
+  // RadixSearch<T, RadixTypeConfig<T>::RadixType, IndexType, false>(
+  RadixSearch<T, typename RadixTypeConfig<T>::RadixType, IndexType, false>(
       cur_input, k, num_cols, shared_mem, &kth_value);
 
   __shared__ int64_t block_min_idx;
@@ -1318,3 +1374,4 @@ bool SortTopk(const phi::GPUContext& dev_ctx,
 }
 }  // namespace funcs
 }  // namespace phi
+//
diff --git a/paddle/phi/kernels/fusion/gpu/fused_dropout_helper.h b/paddle/phi/kernels/fusion/gpu/fused_dropout_helper.h
index 45a29b4cff..8449e3d309 100644
--- a/paddle/phi/kernels/fusion/gpu/fused_dropout_helper.h
+++ b/paddle/phi/kernels/fusion/gpu/fused_dropout_helper.h
@@ -15,7 +15,7 @@
 #pragma once
 
 #if defined(PADDLE_WITH_CUDA)
-#include "paddle/phi/backends/dynload/cublasLt.h"
+// #include "paddle/phi/backends/dynload/cublasLt.h"
 #endif
 
 #include "glog/logging.h"
diff --git a/paddle/phi/kernels/fusion/gpu/fused_layernorm_residual_dropout_bias.h b/paddle/phi/kernels/fusion/gpu/fused_layernorm_residual_dropout_bias.h
index 7d05bcb654..c79cdadabc 100644
--- a/paddle/phi/kernels/fusion/gpu/fused_layernorm_residual_dropout_bias.h
+++ b/paddle/phi/kernels/fusion/gpu/fused_layernorm_residual_dropout_bias.h
@@ -638,9 +638,7 @@ __global__ __launch_bounds__(THREADS_PER_CTA) void fused_fast_ln_fwd_kernel(
         RandVec<VecSize>(&state, rand);
 #pragma unroll
         for (int jt = 0; jt < VecSize; jt++) {
-#ifndef PADDLE_WITH_HIP
-#pragma unroll
-#endif
+// #pragma unroll
           mask_vec[it][jt] = static_cast<MaskType>(rand[jt] >= dropout_prob);
         }
       }
diff --git a/paddle/phi/kernels/fusion/gpu/masked_multihead_attention_kernel.cu b/paddle/phi/kernels/fusion/gpu/masked_multihead_attention_kernel.cu
index ad04265bd6..59481d0e6a 100644
--- a/paddle/phi/kernels/fusion/gpu/masked_multihead_attention_kernel.cu
+++ b/paddle/phi/kernels/fusion/gpu/masked_multihead_attention_kernel.cu
@@ -15,7 +15,7 @@
 #include "paddle/phi/common/bfloat16.h"
 #include "paddle/phi/core/kernel_registry.h"
 #include "paddle/phi/kernels/funcs/aligned_vector.h"
-#include "paddle/phi/kernels/fusion/gpu/mmha_util.cu.h"
+#include "kernels/metax_kernel/mmha_util.cu.h"
 
 namespace phi {
 namespace fusion {
diff --git a/paddle/phi/kernels/fusion/gpu/qkv_unpack_mha_kernel.cu b/paddle/phi/kernels/fusion/gpu/qkv_unpack_mha_kernel.cu
index 148d72ca9c..5da3461ebf 100644
--- a/paddle/phi/kernels/fusion/gpu/qkv_unpack_mha_kernel.cu
+++ b/paddle/phi/kernels/fusion/gpu/qkv_unpack_mha_kernel.cu
@@ -15,7 +15,7 @@
 #include "paddle/phi/common/bfloat16.h"
 #include "paddle/phi/core/kernel_registry.h"
 #include "paddle/phi/kernels/funcs/aligned_vector.h"
-#include "paddle/phi/kernels/fusion/gpu/mmha_util.cu.h"
+#include "kernels/metax_kernel/mmha_util.cu.h"
 
 namespace phi {
 namespace fusion {
diff --git a/paddle/phi/kernels/gpu/depthwise_conv.h b/paddle/phi/kernels/gpu/depthwise_conv.h
index b16553589a..90080c375d 100644
--- a/paddle/phi/kernels/gpu/depthwise_conv.h
+++ b/paddle/phi/kernels/gpu/depthwise_conv.h
@@ -29,8 +29,8 @@ namespace cub = hipcub;
 #include "paddle/phi/backends/gpu/gpu_device_function.h"
 #include "paddle/phi/backends/gpu/gpu_primitives.h"
 #include "paddle/phi/kernels/funcs/math_function.h"
-#include "paddle/phi/kernels/gpudnn/conv_gpudnn.h"
-#include "paddle/phi/kernels/impl/conv_cudnn_impl.h"
+#include "kernels/gpudnn/conv_gpudnn.h"
+#include "kernels/impl/conv_cudnn_impl.h"
 
 namespace phi {
 // To determine use cudnn or not.
diff --git a/paddle/phi/kernels/gpu/gelu_funcs.h b/paddle/phi/kernels/gpu/gelu_funcs.h
index 29fa252e96..4ae72b0935 100644
--- a/paddle/phi/kernels/gpu/gelu_funcs.h
+++ b/paddle/phi/kernels/gpu/gelu_funcs.h
@@ -29,13 +29,13 @@ namespace phi {
 #if defined(__NVCC__) || defined(__HIPCC__)
 template <bool FastMode>
 static __device__ __forceinline__ float FP32FastTanh(float x) {
-#if __CUDA_ARCH__ >= 750 && CUDA_VERSION >= 11000
-  if (FastMode) {
-    float y;
-    asm("tanh.approx.f32 %0,%1; \n\t" : "=f"(y) : "f"(x));
-    return y;
-  }
-#endif
+// #if __CUDA_ARCH__ >= 750 && CUDA_VERSION >= 11000
+//   if (FastMode) {
+//     float y;
+//     asm("tanh.approx.f32 %0,%1; \n\t" : "=f"(y) : "f"(x));
+//     return y;
+//   }
+// #endif
   return tanhf(x);
 }
 
diff --git a/paddle/phi/kernels/gpu/log_softmax_grad_kernel.cu b/paddle/phi/kernels/gpu/log_softmax_grad_kernel.cu
index ee71a2b452..69130ab955 100644
--- a/paddle/phi/kernels/gpu/log_softmax_grad_kernel.cu
+++ b/paddle/phi/kernels/gpu/log_softmax_grad_kernel.cu
@@ -17,7 +17,7 @@
 #include "paddle/phi/backends/gpu/gpu_context.h"
 #include "paddle/phi/core/kernel_registry.h"
 #include "paddle/phi/kernels/funcs/math_function.h"
-#include "paddle/phi/kernels/gpudnn/softmax_gpudnn.h"
+#include "kernels/gpudnn/softmax_gpudnn.h"
 
 namespace phi {
 
diff --git a/paddle/phi/kernels/gpu/log_softmax_kernel.cu b/paddle/phi/kernels/gpu/log_softmax_kernel.cu
index 00a2f1e210..1267cf7ec2 100644
--- a/paddle/phi/kernels/gpu/log_softmax_kernel.cu
+++ b/paddle/phi/kernels/gpu/log_softmax_kernel.cu
@@ -17,7 +17,7 @@
 #include "paddle/phi/backends/gpu/gpu_context.h"
 #include "paddle/phi/core/kernel_registry.h"
 #include "paddle/phi/kernels/funcs/math_function.h"
-#include "paddle/phi/kernels/gpudnn/softmax_gpudnn.h"
+#include "kernels/gpudnn/softmax_gpudnn.h"
 
 namespace phi {
 
diff --git a/paddle/phi/kernels/gpu/lstsq_kernel.cu b/paddle/phi/kernels/gpu/lstsq_kernel.cu
index 1bdbe1564c..f753b54bc6 100644
--- a/paddle/phi/kernels/gpu/lstsq_kernel.cu
+++ b/paddle/phi/kernels/gpu/lstsq_kernel.cu
@@ -21,7 +21,7 @@
 #include "paddle/phi/core/kernel_registry.h"
 #include "paddle/phi/kernels/full_kernel.h"
 #include "paddle/phi/kernels/funcs/slice.h"
-#include "paddle/phi/kernels/impl/lstsq_kernel_impl.h"
+#include "kernels/impl/lstsq_kernel_impl.h"
 #include "paddle/phi/kernels/impl/qr_kernel_impl.h"
 #include "paddle/phi/kernels/impl/tril_triu_kernel_impl.h"
 #include "paddle/phi/kernels/lstsq_kernel.h"
diff --git a/paddle/phi/kernels/impl/addmm_grad_kernel_impl.h b/paddle/phi/kernels/impl/addmm_grad_kernel_impl.h
index 14b24dd3ed..e54a342c98 100644
--- a/paddle/phi/kernels/impl/addmm_grad_kernel_impl.h
+++ b/paddle/phi/kernels/impl/addmm_grad_kernel_impl.h
@@ -21,7 +21,7 @@ limitations under the License. */
 #include "paddle/phi/common/amp_type_traits.h"
 #include "paddle/phi/kernels/addmm_grad_kernel.h"
 #include "paddle/phi/kernels/full_kernel.h"
-#include "paddle/phi/kernels/funcs/blas/blas.h"
+#include "kernels/funcs/blas/blas.h"
 #include "paddle/phi/kernels/funcs/eigen/common.h"
 #include "paddle/phi/kernels/funcs/eigen/eigen_function.h"
 #include "paddle/phi/kernels/funcs/for_range.h"
diff --git a/paddle/phi/kernels/impl/baddbmm_grad_kernel_impl.h b/paddle/phi/kernels/impl/baddbmm_grad_kernel_impl.h
index 06fff0dd58..973049105f 100644
--- a/paddle/phi/kernels/impl/baddbmm_grad_kernel_impl.h
+++ b/paddle/phi/kernels/impl/baddbmm_grad_kernel_impl.h
@@ -19,7 +19,7 @@ limitations under the License. */
 
 #include "paddle/phi/common/amp_type_traits.h"
 #include "paddle/phi/kernels/baddbmm_grad_kernel.h"
-#include "paddle/phi/kernels/funcs/blas/blas.h"
+#include "kernels/funcs/blas/blas.h"
 #include "paddle/phi/kernels/funcs/eigen/common.h"
 #include "paddle/phi/kernels/funcs/eigen/eigen_function.h"
 #include "paddle/phi/kernels/funcs/for_range.h"
diff --git a/paddle/phi/kernels/impl/conv_transpose_grad_kernel_impl.h b/paddle/phi/kernels/impl/conv_transpose_grad_kernel_impl.h
index 9a21c23666..86413d1577 100644
--- a/paddle/phi/kernels/impl/conv_transpose_grad_kernel_impl.h
+++ b/paddle/phi/kernels/impl/conv_transpose_grad_kernel_impl.h
@@ -19,7 +19,7 @@
 #include "paddle/phi/kernels/conv_transpose_grad_kernel.h"
 #include "paddle/phi/kernels/cpu/conv_util.h"
 #include "paddle/phi/kernels/full_kernel.h"
-#include "paddle/phi/kernels/funcs/blas/blas.h"
+#include "kernels/funcs/blas/blas.h"
 #include "paddle/phi/kernels/funcs/concat_and_split_functor.h"
 #include "paddle/phi/kernels/funcs/im2col.h"
 #include "paddle/phi/kernels/funcs/slice.h"
diff --git a/paddle/phi/kernels/impl/deformable_conv_grad_kernel_impl.h b/paddle/phi/kernels/impl/deformable_conv_grad_kernel_impl.h
index 4459a931da..837c8682b8 100644
--- a/paddle/phi/kernels/impl/deformable_conv_grad_kernel_impl.h
+++ b/paddle/phi/kernels/impl/deformable_conv_grad_kernel_impl.h
@@ -18,7 +18,7 @@
 #include "paddle/phi/core/dense_tensor.h"
 #include "paddle/phi/kernels/empty_kernel.h"
 #include "paddle/phi/kernels/full_kernel.h"
-#include "paddle/phi/kernels/funcs/blas/blas.h"
+#include "kernels/funcs/blas/blas.h"
 #include "paddle/phi/kernels/funcs/deformable_conv_functor.h"
 
 namespace phi {
diff --git a/paddle/phi/kernels/impl/gammaincc_kernel_impl.h b/paddle/phi/kernels/impl/gammaincc_kernel_impl.h
index e6b3960f6d..564125f1f6 100644
--- a/paddle/phi/kernels/impl/gammaincc_kernel_impl.h
+++ b/paddle/phi/kernels/impl/gammaincc_kernel_impl.h
@@ -56,8 +56,8 @@ HOSTDEVICE T igam(const T a, const T x) {
 
 template <typename T>
 HOSTDEVICE T igamc(const T a, const T x) {
-  static T big = 4.503599627370496e15;
-  static T biginv = 2.22044604925031308085e-16;
+  const static T big = 4.503599627370496e15;
+  const static T biginv = 2.22044604925031308085e-16;
 
   if ((x <= T{0}) || (a <= T{0})) return (T{1.0});
 
diff --git a/paddle/phi/kernels/impl/gammaln_grad_kernel_impl.h b/paddle/phi/kernels/impl/gammaln_grad_kernel_impl.h
index 410fb3c560..009ce03440 100644
--- a/paddle/phi/kernels/impl/gammaln_grad_kernel_impl.h
+++ b/paddle/phi/kernels/impl/gammaln_grad_kernel_impl.h
@@ -54,7 +54,7 @@ HOSTDEVICE T digamma_positive_domain(T x) {
 
 template <typename T>
 HOSTDEVICE T digamma(T x) {
-  static T pi = T{3.14159265358979323846};
+  const static T pi = T{3.14159265358979323846};
 
   if (x == T{0.0}) {
     T inf = std::numeric_limits<T>::infinity();
diff --git a/paddle/phi/kernels/impl/llm_int8_matmul_kernel_impl.h b/paddle/phi/kernels/impl/llm_int8_matmul_kernel_impl.h
index 5ebbc8d2db..48acf8d0cd 100644
--- a/paddle/phi/kernels/impl/llm_int8_matmul_kernel_impl.h
+++ b/paddle/phi/kernels/impl/llm_int8_matmul_kernel_impl.h
@@ -15,8 +15,9 @@ limitations under the License. */
 #include <iostream>
 #include <vector>
 #include "paddle/phi/common/datatype_traits.h"
-#include "paddle/phi/kernels/funcs/cublaslt.h"
-#include "paddle/phi/kernels/funcs/quant_dequant.h"
+#include "kernels/funcs/blas/cublaslt.h"
+#include "kernels/funcs/quant_dequant.h"
+#include "kernels/metax_context.h"
 
 #pragma once
 
@@ -668,7 +669,7 @@ void LLMGemm(const phi::GPUContext& dev_ctx,
 
   {
     auto helper =
-        std::make_unique<CublasLtHelper>(m, k, n, dev_ctx.cublaslt_handle());
+        std::make_unique<CublasLtHelper>(m, k, n, GetBlasLtHandle());
     helper->GEMM(quant_input.data<int8_t>(),
                  weight->data<int8_t>(),
                  int_out.data<int32_t>(),
diff --git a/third_party/yaml-cpp b/third_party/yaml-cpp
--- a/third_party/yaml-cpp
+++ b/third_party/yaml-cpp
@@ -1 +1 @@
-Subproject commit 1d8ca1f35eb3a9c9142462b28282a848e5d29a91
+Subproject commit 1d8ca1f35eb3a9c9142462b28282a848e5d29a91-dirty
diff --git a/paddle/phi/kernels/impl/baddbmm_kernel_impl.h b/paddle/phi/kernels/impl/baddbmm_kernel_impl.h
index 2789cb59a2..b91b076f7f 100644
--- a/paddle/phi/kernels/impl/baddbmm_kernel_impl.h
+++ b/paddle/phi/kernels/impl/baddbmm_kernel_impl.h
@@ -20,7 +20,7 @@ limitations under the License. */
 
 #include "paddle/phi/common/amp_type_traits.h"
 #include "paddle/phi/kernels/baddbmm_kernel.h"
-#include "paddle/phi/kernels/funcs/blas/blas.h"
+#include "kernels/funcs/blas/blas.h"
 #include "paddle/phi/kernels/funcs/eigen/common.h"
 #include "paddle/phi/kernels/funcs/eigen/eigen_function.h"
 
