diff --git a/cmake/external/eigen.cmake b/cmake/external/eigen.cmake
index 6790d961f8..37b841bfc4 100644
--- a/cmake/external/eigen.cmake
+++ b/cmake/external/eigen.cmake
@@ -42,14 +42,14 @@ endif()
 file(TO_NATIVE_PATH "${PADDLE_SOURCE_DIR}/patches/eigen/TensorRandom.h.patch"
      tensor_random_header)
 # See: [Why calling some `git` commands before `patch`?]
-set(EIGEN_PATCH_COMMAND
-    git checkout -- Eigen/src/Core/arch/SSE/Complex.h
-    unsupported/Eigen/CXX11/src/Tensor/TensorRandom.h && git checkout
-    ${EIGEN_TAG} && git apply ${tensor_random_header})
+# set(EIGEN_PATCH_COMMAND
+#     git checkout -- Eigen/src/Core/arch/SSE/Complex.h
+#     unsupported/Eigen/CXX11/src/Tensor/TensorRandom.h && git checkout
+#     ${EIGEN_TAG} && git apply ${tensor_random_header})
 if(CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang")
   file(TO_NATIVE_PATH ${PADDLE_SOURCE_DIR}/patches/eigen/Complex.h.patch
        complex_header)
-  set(EIGEN_PATCH_COMMAND ${EIGEN_PATCH_COMMAND} && git apply ${complex_header})
+  # set(EIGEN_PATCH_COMMAND ${EIGEN_PATCH_COMMAND} && git apply ${complex_header})
 endif()
 
 set(EIGEN_INCLUDE_DIR ${SOURCE_DIR})
diff --git a/paddle/fluid/operators/fused/CMakeLists.txt b/paddle/fluid/operators/fused/CMakeLists.txt
index 8d445b39ae..504e7b6293 100755
--- a/paddle/fluid/operators/fused/CMakeLists.txt
+++ b/paddle/fluid/operators/fused/CMakeLists.txt
@@ -39,6 +39,11 @@ if(WITH_GPU OR WITH_ROCM)
     op_library(fused_multi_transformer_int8_op)
   endif()
 
+  if 1
+  op_library(fused_gemm_epilogue_op)
+  endif()
+
+
   if(CUDA_VERSION GREATER_EQUAL 11.6)
     op_library(fused_gemm_epilogue_op)
   endif()
diff --git a/paddle/phi/backends/dynload/dynamic_loader.cc b/paddle/phi/backends/dynload/dynamic_loader.cc
index 99d3733da0..d0c8783afd 100644
--- a/paddle/phi/backends/dynload/dynamic_loader.cc
+++ b/paddle/phi/backends/dynload/dynamic_loader.cc
@@ -19,7 +19,7 @@ limitations under the License. */
 #include <locale>
 #include <string>
 #include <vector>
-#include "paddle/phi/backends/dynload/cupti_lib_path.h"
+// #include "paddle/phi/backends/dynload/cupti_lib_path.h"
 #include "paddle/phi/common/port.h"
 #include "paddle/phi/core/enforce.h"
 
@@ -113,6 +113,10 @@ COMMON_DECLARE_string(magma_dir);
 #define SPARSELT_LIB_NAME "libcusparseLt.so"
 #endif
 
+#ifndef CUPTI_LIB_PATH
+#define CUPTI_LIB_PATH "@CUPTI_LIBRARY_PATH@"
+#endif
+
 #ifdef PADDLE_WITH_HIP
 
 PHI_DEFINE_string(miopen_dir,
diff --git a/paddle/phi/backends/gpu/cuda/cuda_device_function.h b/paddle/phi/backends/gpu/cuda/cuda_device_function.h
index 092365a961..8bd3f9fcea 100644
--- a/paddle/phi/backends/gpu/cuda/cuda_device_function.h
+++ b/paddle/phi/backends/gpu/cuda/cuda_device_function.h
@@ -1,3 +1,4 @@
+// 2024 - Modified by MetaX Integrated Circuits (Shanghai) Co., Ltd. All Rights Reserved.
 /* Copyright (c) 2022 PaddlePaddle Authors. All Rights Reserved.
 
 Licensed under the Apache License, Version 2.0 (the "License");
@@ -23,7 +24,7 @@ namespace phi {
 namespace backends {
 namespace gpu {
 
-#define FULL_WARP_MASK 0xFFFFFFFF
+#define FULL_WARP_MASK 0xFFFFFFFFFFFFFFFFULL
 #define CREATE_SHFL_MASK(mask, predicate) \
   mask = __ballot_sync(FULL_WARP_MASK, (predicate))
 
@@ -43,12 +44,12 @@ namespace gpu {
 
 template <typename T>
 __forceinline__ __device__ T
-CudaShuffleDownSync(unsigned mask, T val, int delta, int width = warpSize) {
+CudaShuffleDownSync(unsigned long long mask, T val, int delta, int width = warpSize) {
   return __shfl_down_sync(mask, val, static_cast<unsigned>(delta), width);
 }
 
 template <typename T>
-__forceinline__ __device__ T CudaShuffleXorSync(unsigned mask,
+__forceinline__ __device__ T CudaShuffleXorSync(unsigned long long mask,
                                                 T val,
                                                 int width = warpSize) {
   return __shfl_xor_sync(mask, val, width);
@@ -56,14 +57,14 @@ __forceinline__ __device__ T CudaShuffleXorSync(unsigned mask,
 
 template <>
 __forceinline__ __device__ phi::dtype::float16 CudaShuffleDownSync(
-    unsigned mask, phi::dtype::float16 val, int delta, int width) {
+    unsigned long long mask, phi::dtype::float16 val, int delta, int width) {
   return phi::dtype::float16(__shfl_down_sync(
       mask, val.to_half(), static_cast<unsigned>(delta), width));
 }
 
 template <>
 __forceinline__ __device__ phi::dtype::bfloat16 CudaShuffleDownSync(
-    unsigned mask, phi::dtype::bfloat16 val, int delta, int width) {
+    unsigned long long mask, phi::dtype::bfloat16 val, int delta, int width) {
 #if defined(PADDLE_CUDA_BF16)
   return phi::dtype::bfloat16(__shfl_down_sync(
       mask, val.to_nv_bfloat16(), static_cast<unsigned>(delta), width));
@@ -75,7 +76,7 @@ __forceinline__ __device__ phi::dtype::bfloat16 CudaShuffleDownSync(
 
 template <>
 __forceinline__ __device__ phi::dtype::complex<float> CudaShuffleDownSync(
-    unsigned mask, phi::dtype::complex<float> val, int delta, int width) {
+    unsigned long long mask, phi::dtype::complex<float> val, int delta, int width) {
   float real = static_cast<float>(__shfl_down_sync(
       mask, static_cast<float>(val.real), static_cast<unsigned>(delta), width));
   float imag = static_cast<float>(__shfl_down_sync(
@@ -85,7 +86,7 @@ __forceinline__ __device__ phi::dtype::complex<float> CudaShuffleDownSync(
 
 template <>
 __forceinline__ __device__ phi::dtype::complex<double> CudaShuffleDownSync(
-    unsigned mask, phi::dtype::complex<double> val, int delta, int width) {
+    unsigned long long mask, phi::dtype::complex<double> val, int delta, int width) {
   double real =
       static_cast<double>(__shfl_down_sync(mask,
                                            static_cast<double>(val.real),
@@ -101,20 +102,20 @@ __forceinline__ __device__ phi::dtype::complex<double> CudaShuffleDownSync(
 
 template <>
 __forceinline__ __device__ phi::dtype::float16 CudaShuffleXorSync(
-    unsigned mask, phi::dtype::float16 val, int width) {
+    unsigned long long mask, phi::dtype::float16 val, int width) {
   return phi::dtype::float16(__shfl_xor_sync(mask, val.to_half(), width));
 }
 
 template <>
 __forceinline__ __device__ phi::dtype::bfloat16 CudaShuffleXorSync(
-    unsigned mask, phi::dtype::bfloat16 val, int width) {
+    unsigned long long mask, phi::dtype::bfloat16 val, int width) {
   return phi::dtype::bfloat16(
       __shfl_xor_sync(mask, val.to_nv_bfloat16(), width));
 }
 
 template <>
 __forceinline__ __device__ phi::dtype::complex<float> CudaShuffleXorSync(
-    unsigned mask, phi::dtype::complex<float> val, int width) {
+    unsigned long long mask, phi::dtype::complex<float> val, int width) {
   float real = static_cast<float>(
       __shfl_xor_sync(mask, static_cast<float>(val.real), width));
   float imag = static_cast<float>(
@@ -124,7 +125,7 @@ __forceinline__ __device__ phi::dtype::complex<float> CudaShuffleXorSync(
 
 template <>
 __forceinline__ __device__ phi::dtype::complex<double> CudaShuffleXorSync(
-    unsigned mask, phi::dtype::complex<double> val, int width) {
+    unsigned long long mask, phi::dtype::complex<double> val, int width) {
   double real = static_cast<double>(
       __shfl_xor_sync(mask, static_cast<double>(val.real), width));
   double imag = static_cast<double>(
@@ -134,7 +135,7 @@ __forceinline__ __device__ phi::dtype::complex<double> CudaShuffleXorSync(
 
 template <typename T>
 __forceinline__ __device__ T
-CudaShuffleSync(unsigned mask, T val, int src_line, int width = 32) {
+CudaShuffleSync(unsigned long long mask, T val, int src_line, int width = 32) {
   return __shfl_sync(mask, val, src_line, width);
 }
 
@@ -153,7 +154,7 @@ __device__ T reduceSum(T val, int tid, int len) {
   // but most card's warp size is 32.
   const int warpSize = 32;
   __shared__ T shm[warpSize];
-  unsigned mask = 0u;
+  unsigned long long mask = 0ull;
   CREATE_SHFL_MASK(mask, tid < len);
 
   for (int offset = warpSize / 2; offset > 0; offset /= 2)
diff --git a/paddle/phi/common/float16.h b/paddle/phi/common/float16.h
index d970878dc2..fe0382ccad 100644
--- a/paddle/phi/common/float16.h
+++ b/paddle/phi/common/float16.h
@@ -105,8 +105,9 @@ struct PADDLE_ALIGN(2) float16 {
 #endif
 
   HOSTDEVICE inline explicit float16(float val) {
-#if defined(PADDLE_CUDA_FP16) && \
-    (defined(__HIPCC__) || (defined(__CUDA_ARCH__) && __CUDA_ARCH__ >= 300))
+// #if defined(PADDLE_CUDA_FP16) && \
+//     (defined(__HIPCC__) || (defined(__CUDA_ARCH__) && __CUDA_ARCH__ >= 300))
+#if 1
     half tmp = __float2half(val);
     x = *reinterpret_cast<uint16_t*>(&tmp);
 
diff --git a/paddle/phi/core/enforce.h b/paddle/phi/core/enforce.h
index 024a7de73e..66b373d698 100644
--- a/paddle/phi/core/enforce.h
+++ b/paddle/phi/core/enforce.h
@@ -97,7 +97,7 @@ inline bool is_error(bool stat) { return !stat; }
 
 void ThrowWarnInternal(const std::string& message);
 
-#if defined(__CUDA_ARCH__)
+#if defined(__CUDACC__)
 // For cuda, the assertions can affect performance and it is therefore
 // recommended to disable them in production code
 // https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#assertion
@@ -109,7 +109,7 @@ void ThrowWarnInternal(const std::string& message);
              __LINE__,                                             \
              #_IS_NOT_ERROR,                                       \
              ##__VA_ARGS__);                                       \
-      asm("trap;");                                                \
+      __builtin_trap();                                            \
     }                                                              \
   } while (0)
 #elif defined(__HIPCC__)
diff --git a/paddle/phi/kernels/funcs/blas/blas_impl.cu.h b/paddle/phi/kernels/funcs/blas/blas_impl.cu.h
index fc86cc09d1..037f020d00 100644
--- a/paddle/phi/kernels/funcs/blas/blas_impl.cu.h
+++ b/paddle/phi/kernels/funcs/blas/blas_impl.cu.h
@@ -219,11 +219,27 @@ struct CUBlas<float> {
   }
 };
 
+template<typename... Args>
+void print_args(Args... args) {
+    std::cout << "Arguments (" << sizeof...(args) << "): [";
+    bool first = true;
+    auto printer = [&first](const auto& arg) {
+        if (!first) std::cout << ", ";
+        std::cout << arg;
+        first = false;
+    };
+    (printer(args), ...);
+    std::cout << "]" << std::endl;
+}
+
 template <>
 struct CUBlas<double> {
   template <typename... ARGS>
   static void GEMM(ARGS... args) {
+    // print_args(args...);
     PADDLE_ENFORCE_GPU_SUCCESS(phi::dynload::cublasDgemm(args...));
+    
+    
   }
 
   template <typename... ARGS>
@@ -369,7 +385,7 @@ struct CUBlas<phi::float16> {
                          cudaDataType_t Ctype,
                          int ldc,
                          int batchCount,
-                         cudaDataType_t computeType) {
+                         cublasComputeType_t computeType) {
 #if CUDA_VERSION >= 8000
     cublasGemmAlgo_t algo = CUBLAS_GEMM_DFALT;
 #if CUDA_VERSION >= 9000
@@ -477,7 +493,7 @@ struct CUBlas<phi::float16> {
                       void *C,
                       cudaDataType_t Ctype,
                       int ldc,
-                      cudaDataType_t computeType) {
+                      cublasComputeType_t computeType) {
 #if CUDA_VERSION >= 8000
     cublasGemmAlgo_t algo = CUBLAS_GEMM_DFALT;
 #if CUDA_VERSION >= 9000
@@ -533,7 +549,7 @@ struct CUBlas<phi::float16> {
                          void *C,
                          cudaDataType_t Ctype,
                          int64_t ldc,
-                         cudaDataType_t computeType) {
+                         cublasComputeType_t computeType) {
 #if CUDA_VERSION >= 12030 && defined(__linux__)
     cublasGemmAlgo_t algo = CUBLAS_GEMM_DFALT;
     bool use_tensor_op_math = dev_ctx->tensor_core_available();
@@ -760,7 +776,7 @@ struct CUBlas<phi::complex64> {
                       void *C,
                       cudaDataType_t Ctype,
                       int ldc,
-                      cudaDataType_t computeType) {
+                      cublasComputeType_t computeType) {
 #if CUDA_VERSION >= 8000
     cublasGemmAlgo_t algo = CUBLAS_GEMM_DFALT;
 #if CUDA_VERSION >= 9000
@@ -816,7 +832,7 @@ struct CUBlas<phi::complex64> {
                          void *C,
                          cudaDataType_t Ctype,
                          int64_t ldc,
-                         cudaDataType_t computeType) {
+                         cublasComputeType_t computeType) {
 #if CUDA_VERSION >= 12030 && defined(__linux__)
     cublasGemmAlgo_t algo = CUBLAS_GEMM_DFALT;
     bool use_tensor_op_math = dev_ctx->tensor_core_available();
@@ -1155,7 +1171,7 @@ struct CUBlas<phi::complex128> {
                       void *C,
                       cudaDataType_t Ctype,
                       int ldc,
-                      cudaDataType_t computeType) {
+                      cublasComputeType_t computeType) {
 #if CUDA_VERSION >= 8000
     cublasGemmAlgo_t algo = CUBLAS_GEMM_DFALT;
 #if CUDA_VERSION >= 9000
@@ -1211,7 +1227,7 @@ struct CUBlas<phi::complex128> {
                          void *C,
                          cudaDataType_t Ctype,
                          int64_t ldc,
-                         cudaDataType_t computeType) {
+                         cublasComputeType_t computeType) {
 #if CUDA_VERSION >= 12030 && defined(__linux__)
     cublasGemmAlgo_t algo = CUBLAS_GEMM_DFALT;
     bool use_tensor_op_math = dev_ctx->tensor_core_available();
@@ -1485,7 +1501,7 @@ inline void Blas<phi::GPUContext>::GEMM(CBLAS_TRANSPOSE transA,
                                      C,
                                      CUDA_R_16F,
                                      N,
-                                     CUDA_R_32F);
+                                     CUBLAS_COMPUTE_32F);
 #else
     PADDLE_THROW(common::errors::Unimplemented(
         "GEMM_EX_64 is not supported on cuda < 12.3"));
@@ -1509,7 +1525,7 @@ inline void Blas<phi::GPUContext>::GEMM(CBLAS_TRANSPOSE transA,
                                   C,
                                   CUDA_R_16F,
                                   static_cast<int>(N),
-                                  CUDA_R_32F);
+                                  CUBLAS_COMPUTE_32F);
   }
 #else
   // CUDA 7.5 does not support cublasGemmEx, hence we fall back to use hgemm
@@ -1695,7 +1711,7 @@ inline void Blas<phi::GPUContext>::GEMM(CBLAS_TRANSPOSE transA,
                                      C,
                                      CUDA_R_16F,
                                      N,
-                                     CUDA_R_32F);
+                                     CUBLAS_COMPUTE_32F);
 #else
     PADDLE_THROW(common::errors::Unimplemented(
         "GEMM_EX_64 is not supported on cuda < 12.3"));
@@ -1720,7 +1736,7 @@ inline void Blas<phi::GPUContext>::GEMM(CBLAS_TRANSPOSE transA,
                                   C,
                                   CUDA_R_16F,
                                   static_cast<int>(N),
-                                  CUDA_R_32F);
+                                  CUBLAS_COMPUTE_32F);
 #else
     // CUDA 7.5 does not support cublasGemmEx, hence we fall back to use hgemm
     dev_ctx_.CublasCall([&](cublasHandle_t handle) {
@@ -1832,7 +1848,7 @@ inline void Blas<phi::GPUContext>::GEMM(CBLAS_TRANSPOSE transA,
                                      C,
                                      CUDA_R_16BF,
                                      static_cast<int>(N),
-                                     CUDA_R_32F,
+                                     CUBLAS_COMPUTE_32F,
                                      algo));
     });
   }
@@ -1933,7 +1949,7 @@ inline void Blas<phi::GPUContext>::GEMM(CBLAS_TRANSPOSE transA,
                                      C,
                                      CUDA_R_16BF,
                                      static_cast<int>(N),
-                                     CUDA_R_32F,
+                                     CUBLAS_COMPUTE_32F,
                                      algo));
     });
   }
@@ -2027,7 +2043,7 @@ inline void Blas<phi::GPUContext>::GEMM(CBLAS_TRANSPOSE transA,
                                     C,
                                     CUDA_C_32F,
                                     static_cast<int>(N),
-                                    CUDA_C_32F);
+                                    CUBLAS_COMPUTE_32F);
 
 #else
     dev_ctx_.CublasCall([&](cublasHandle_t handle) {
@@ -2112,7 +2128,7 @@ inline void Blas<phi::GPUContext>::GEMM(CBLAS_TRANSPOSE transA,
                                         C,
                                         CUDA_C_64F,
                                         N,
-                                        CUDA_C_64F);
+                                        CUBLAS_COMPUTE_64F);
 #else
     PADDLE_THROW(common::errors::Unimplemented(
         "GEMM_EX_64 is not supported on cuda < 12.3"));
@@ -2137,7 +2153,7 @@ inline void Blas<phi::GPUContext>::GEMM(CBLAS_TRANSPOSE transA,
                                      C,
                                      CUDA_C_64F,
                                      static_cast<int>(N),
-                                     CUDA_C_64F);
+                                     CUBLAS_COMPUTE_64F);
 #else  // CUDA_VERSION >= 8000
     // CUDA 7.5 does not support cublasGemmEx, hence we fall back to use hgemm
     dev_ctx_.CublasCall([&](cublasHandle_t handle) {
@@ -2273,7 +2289,7 @@ inline void Blas<phi::GPUContext>::GEMM(bool transA,
                                                           C,
                                                           CUDA_R_16F,
                                                           ldc,
-                                                          CUDA_R_32F,
+                                                          CUBLAS_COMPUTE_32F,
                                                           algo));
   });
 }
@@ -2335,7 +2351,7 @@ inline void Blas<phi::GPUContext>::GEMM(bool transA,
                                                           C,
                                                           CUDA_R_16BF,
                                                           ldc,
-                                                          CUDA_R_32F,
+                                                          CUBLAS_COMPUTE_32F,
                                                           algo));
   });
 #else
@@ -3159,7 +3175,7 @@ inline void Blas<phi::GPUContext>::BatchedGEMM(CBLAS_TRANSPOSE transA,
                                    CUDA_R_16F,
                                    ldc,
                                    batchCount,
-                                   CUDA_R_32F);
+                                   CUBLAS_COMPUTE_32F);
 }
 
 template <>
@@ -3227,7 +3243,7 @@ inline void Blas<phi::GPUContext>::BatchedGEMM(CBLAS_TRANSPOSE transA,
                                           CUDA_R_16BF,
                                           ldc,
                                           batchCount,
-                                          CUDA_R_32F,
+                                          CUBLAS_COMPUTE_32F,
                                           algo));
   });
 #else
diff --git a/paddle/phi/kernels/funcs/blas/blaslt_gemm_search.h b/paddle/phi/kernels/funcs/blas/blaslt_gemm_search.h
index aab18dd414..683c323f53 100644
--- a/paddle/phi/kernels/funcs/blas/blaslt_gemm_search.h
+++ b/paddle/phi/kernels/funcs/blas/blaslt_gemm_search.h
@@ -628,7 +628,13 @@ class CublasLtAlgoCache {
     infile >> cublaslt_version;
     VLOG(1) << "cublaslt_version " << cublaslt_version;
 
-    if (dynload::cublasLtGetCudartVersion() != cublaslt_version) {
+    // if (dynload::cublasLtGetCudartVersion() != cublaslt_version) {
+    //   LOG(INFO) << algo_caches_file_
+    //             << " is not compatible with current cublaslt_version "
+    //             << real_cublaslt_version;
+    //   return;
+    // }
+    if (3000 != cublaslt_version) {
       LOG(INFO) << algo_caches_file_
                 << " is not compatible with current cublaslt_version "
                 << real_cublaslt_version;
@@ -655,7 +661,8 @@ class CublasLtAlgoCache {
       if (dev == 0) {
         std::ofstream outfile;
         outfile.open(algo_caches_file_, std::ios::out | std::ios::trunc);
-        outfile << dynload::cublasLtGetCudartVersion() << std::endl;
+        // outfile << dynload::cublasLtGetCudartVersion() << std::endl;
+        outfile << 3000 << std::endl;
 
         for (const auto& [seed, algo] : algo_caches_) {
           outfile << seed << " ";
diff --git a/paddle/phi/kernels/funcs/embedding_grad.h b/paddle/phi/kernels/funcs/embedding_grad.h
index 461e6e2474..48a64ae9ce 100644
--- a/paddle/phi/kernels/funcs/embedding_grad.h
+++ b/paddle/phi/kernels/funcs/embedding_grad.h
@@ -143,8 +143,8 @@ void LaunchEmbeddingGradDeterministicKernel(const GPUContext& dev_ctx,
   constexpr int kWarpSize = 64;
   constexpr int kBlockDimY = 16;
 #else
-  constexpr int kWarpSize = 32;
-  constexpr int kBlockDimY = 32;
+  constexpr int kWarpSize = 64;
+  constexpr int kBlockDimY = 16;
 #endif
   dim3 threads(kWarpSize, kBlockDimY);
   dim3 grids(static_cast<int>((D + kWarpSize - 1) / kWarpSize));
diff --git a/paddle/phi/kernels/funcs/fast_ln_v2_utils.h b/paddle/phi/kernels/funcs/fast_ln_v2_utils.h
index 6435688914..f81f5246af 100644
--- a/paddle/phi/kernels/funcs/fast_ln_v2_utils.h
+++ b/paddle/phi/kernels/funcs/fast_ln_v2_utils.h
@@ -296,12 +296,9 @@ struct InterCTASync {
 
   inline __device__ void spin_wait_(int *barrier, int step, int expected) {
 #if defined(__CUDA_ARCH__) && (__CUDA_ARCH__ >= 700)
-    asm volatile("red.release.gpu.global.add.s32 [%0], %1;" ::"l"(barrier),
-                 "r"(step));
-    for (int found = -1; found != expected;) {
-      asm volatile("ld.global.acquire.gpu.b32 %0, [%1];"
-                   : "=r"(found)
-                   : "l"(barrier));
+    atomicAdd(barrier, step);
+    volatile int* vbarrier = reinterpret_cast<volatile int*>(barrier);
+    while (vbarrier[0] != expected) {
     }
 #endif
   }
diff --git a/paddle/phi/kernels/funcs/fft.cu b/paddle/phi/kernels/funcs/fft.cu
index da92193d10..ae3b84f9be 100644
--- a/paddle/phi/kernels/funcs/fft.cu
+++ b/paddle/phi/kernels/funcs/fft.cu
@@ -172,7 +172,7 @@ void exec_fft(const phi::GPUContext& dev_ctx,
   int64_t device_id = dev_ctx.GetPlace().GetDeviceId();
   FFTConfig* config = nullptr;
   std::unique_ptr<FFTConfig> config_ = nullptr;
-  bool using_cache = use_cache(key.sizes_);
+  bool using_cache = false;//use_cache(key.sizes_);
 
   if (using_cache) {
     FFTConfigCache& plan_cache = get_fft_plan_cache(device_id);
diff --git a/paddle/phi/kernels/funcs/layer_norm_impl.cu.h b/paddle/phi/kernels/funcs/layer_norm_impl.cu.h
index 93540f1df6..5fa8d3ae4a 100644
--- a/paddle/phi/kernels/funcs/layer_norm_impl.cu.h
+++ b/paddle/phi/kernels/funcs/layer_norm_impl.cu.h
@@ -36,11 +36,11 @@ template <typename T>
 using LayerNormParamType = typename CudnnDataType<T>::BatchNormParamType;
 
 inline static int GetDesiredBlockDim(int64_t block_dim) {
-  const int kMaxBlockDim = 512;
+  const int kMaxBlockDim = 256;
 #ifdef __HIPCC__
   const int lwarpSize = 64;
 #else
-  const int lwarpSize = 32;
+  const int lwarpSize = 64;
 #endif
   return block_dim >= kMaxBlockDim ? kMaxBlockDim : lwarpSize;
 }
diff --git a/paddle/phi/kernels/funcs/math_cuda_utils.h b/paddle/phi/kernels/funcs/math_cuda_utils.h
index dff1033db4..0098123818 100644
--- a/paddle/phi/kernels/funcs/math_cuda_utils.h
+++ b/paddle/phi/kernels/funcs/math_cuda_utils.h
@@ -175,12 +175,12 @@ struct KeyValuePair<half> {
 #define WARP_SIZE_WIDTH_MASK 0x3f
 typedef u_int64_t warp_mask_t;
 #else
-#define FINAL_MASK 0xffffffff
-#define HALF_WARP 16
-#define WARP_SIZE 32
-#define WARP_SIZE_WIDTH 5
-#define WARP_SIZE_WIDTH_MASK 0x1f
-typedef unsigned warp_mask_t;
+#define FINAL_MASK 0xffffffffffffffffUL
+#define HALF_WARP 32
+#define WARP_SIZE 64
+#define WARP_SIZE_WIDTH 6
+#define WARP_SIZE_WIDTH_MASK 0x3f
+typedef u_int64_t warp_mask_t;
 #endif
 
 template <typename T>
@@ -200,19 +200,13 @@ __inline__ __device__ T BlockReduceSum(T val, warp_mask_t mask) {
   static __shared__ T shared[WARP_SIZE];
   int lane = threadIdx.x & WARP_SIZE_WIDTH_MASK;
   int wid = threadIdx.x >> WARP_SIZE_WIDTH;
-
   val = WarpReduceSum<T>(val, mask);
-
-  __syncthreads();
   if (lane == 0) shared[wid] = val;
-
   __syncthreads();
-
   // align block_span to warpSize
   int block_span = (blockDim.x + warpSize - 1) >> WARP_SIZE_WIDTH;
   val = (lane < block_span) ? shared[lane] : static_cast<T>(0.0f);
   val = WarpReduceSum<T>(val, mask);
-
   return val;
 }
 
diff --git a/paddle/phi/kernels/funcs/top_k_function_cuda.h b/paddle/phi/kernels/funcs/top_k_function_cuda.h
index 13670ffc90..f47ee23d31 100644
--- a/paddle/phi/kernels/funcs/top_k_function_cuda.h
+++ b/paddle/phi/kernels/funcs/top_k_function_cuda.h
@@ -30,11 +30,11 @@ limitations under the License. */
 #include "paddle/phi/kernels/funcs/eigen/eigen_function.h"
 #include "paddle/phi/kernels/primitive/functor_primitives.h"
 
-#define FINAL_MASK 0xffffffff
+#define FINAL_MASK 0xffffffffffffffffull
 #ifdef PADDLE_WITH_HIP
 #define WARP_SIZE 64
 #else
-#define WARP_SIZE 32
+#define WARP_SIZE 64
 #endif
 #define MAX_NUM_THREADS 1024
 
@@ -196,21 +196,56 @@ __device__ __forceinline__ void AddTo(Pair<T> topk[],
   for (int k = beam_size - 2; k >= 0; k--) {
     if (largest) {
       if (topk[k] < p) {
-        topk[k + 1] = topk[k];
+        topk[k + 1].v = topk[k].v;
+        topk[k + 1].id = topk[k].id;
       } else {
-        topk[k + 1] = p;
+        topk[k + 1].v = p.v;
+        topk[k + 1].id = p.id;
         return;
       }
     } else {
       if (topk[k] > p) {
-        topk[k + 1] = topk[k];
+        topk[k + 1].v = topk[k].v;
+        topk[k + 1].id = topk[k].id;
       } else {
-        topk[k + 1] = p;
+        topk[k + 1].v = p.v;
+        topk[k + 1].id = p.id;
         return;
       }
     }
   }
-  topk[0] = p;
+  topk[0].v = p.v;
+  topk[0].id = p.id;
+}
+
+template <typename T>
+__device__ __forceinline__ void AddTo(Pair<T> topk[],
+                                      const Pair<T>& p,
+                                      int beam_size,
+                                      const bool& largest, const int offset) {
+  for (int k = beam_size - 2; k >= 0; k--) {
+    if (largest) {
+      if (topk[k + offset] < p) {
+        topk[k + 1 + offset].v = topk[k + offset].v;
+        topk[k + 1 + offset].id = topk[k + offset].id;
+      } else {
+        topk[k + 1 + offset].v = p.v;
+        topk[k + 1 + offset].id = p.id;
+        return;
+      }
+    } else {
+      if (topk[k + offset] > p) {
+        topk[k + 1 + offset].v = topk[k + offset].v;
+        topk[k + 1 + offset].id = topk[k + offset].id;
+      } else {
+        topk[k + 1 + offset].v = p.v;
+        topk[k + 1 + offset].id = p.id;
+        return;
+      }
+    }
+  }
+  topk[0 + offset].v = p.v;
+  topk[0 + offset].id = p.id;
 }
 
 template <typename T, int BlockSize>
@@ -239,24 +274,24 @@ __device__ __forceinline__ void GetTopK(Pair<T> topk[],
 template <typename T, int BlockSize>
 __device__ __forceinline__ void GetTopK(Pair<T> topk[],
                                         const T* src,
-                                        int64_t idx,
-                                        int64_t dim,
+                                        int idx,
+                                        int dim,
                                         const Pair<T>& max,
                                         int beam_size,
-                                        const bool& largest) {
+                                        const bool& largest, const int offset) {
   while (idx < dim) {
     if (largest) {
-      if (topk[beam_size - 1] < src[idx]) {
+      if (topk[beam_size - 1 + offset] < src[idx]) {
         Pair<T> tmp(src[idx], idx);
         if (tmp < max) {
-          AddTo<T>(topk, tmp, beam_size, largest);
+          AddTo<T>(topk, tmp, beam_size, largest,offset);
         }
       }
     } else {
-      if (topk[beam_size - 1] > src[idx]) {
+      if (topk[beam_size - 1 + offset] > src[idx]) {
         Pair<T> tmp(src[idx], idx);
         if (tmp > max) {
-          AddTo<T>(topk, tmp, beam_size, largest);
+          AddTo<T>(topk, tmp, beam_size, largest,offset);
         }
       }
     }
@@ -283,7 +318,9 @@ __device__ __forceinline__ void ThreadGetTopK(Pair<T> topk[],
     } else {
       for (int k = 0; k < MaxLength; k++) {
         if (k < MaxLength - (*beam)) {
-          topk[k] = topk[k + *beam];
+          topk[k].v = topk[k + *beam].v;
+          topk[k].id = topk[k + *beam].id;
+
         } else {
           if (largest) {
             topk[k].set(-static_cast<T>(INFINITY), -1);
@@ -293,8 +330,10 @@ __device__ __forceinline__ void ThreadGetTopK(Pair<T> topk[],
         }
       }
       if (!(*is_empty)) {
+        const int offset =  MaxLength - *beam;
         GetTopK<T, BlockSize>(
-            topk + MaxLength - *beam, src, tid, dim, *max, length, largest);
+            topk, src, tid, dim, *max, length, largest, offset);
+            // topk + MaxLength - *beam, src, tid, dim, *max, length, largest);
       }
     }
 
@@ -355,6 +394,8 @@ __device__ __forceinline__ void BlockReduce(Pair<T> shared_max[],
       shared_max[wid] = input_now;
     }
     __syncthreads();
+    if(BlockSize > WARP_SIZE)
+    {
     if (largest) {
       input_now = (tid < BlockSize / WARP_SIZE)
                       ? shared_max[lane]
@@ -369,27 +410,32 @@ __device__ __forceinline__ void BlockReduce(Pair<T> shared_max[],
       if (lane == 0) shared_max[0] = input_now;
     }
     __syncthreads();
-
+    }
     if (tid == 0) {
       **topVal = input_now.v;
       **topIds = input_now.id;
       (*topVal)++;
       (*topIds)++;
     }
+    if (--(*k) == 0) break;
     int tid_max = shared_max[0].id % BlockSize;
     if (tid == tid_max) {
       (*beam)++;
       if (*beam < MaxLength) {
-        topk[0] = topk[*beam];
+        topk[0].v = topk[*beam].v;
+        topk[0].id = topk[*beam].id;
       }
     }
-    if (--(*k) == 0) break;
+    // if (--(*k) == 0) break;
+    unsigned long long mask = 0ull;
 
-    unsigned mask = 0u;
+    // unsigned mask = 0u;
     CREATE_SHFL_MASK(mask, true);
     if (tid_max / WARP_SIZE == wid) {
-      if (phi::backends::gpu::CudaShuffleSync(
-              mask, *beam, tid_max % WARP_SIZE, WARP_SIZE) == MaxLength)
+      // if (phi::backends::gpu::CudaShuffleSync(
+      //         mask, *beam, tid_max % WARP_SIZE, WARP_SIZE) == MaxLength)
+    if (phi::backends::gpu::CudaShuffleSync(
+                 mask, *beam, tid_max % WARP_SIZE, WARP_SIZE) == MaxLength)
         break;
     }
   }
@@ -478,16 +524,17 @@ struct Bitfield<unsigned int> {
                                                              int pos,
                                                              int len) {
     unsigned int ret;
-    asm("bfe.u32 %0, %1, %2, %3;" : "=r"(ret) : "r"(val), "r"(pos), "r"(len));
+  ret = (static_cast<unsigned int>(val) << (32 - pos - len)) >> (32 - len);
+
+
     return ret;
   }
 
   static __device__ __forceinline__ unsigned int SetBitfield(
       unsigned int val, unsigned int to_insert, int pos, int len) {
     unsigned int ret;
-    asm("bfi.b32 %0, %1, %2, %3, %4;"
-        : "=r"(ret)
-        : "r"(to_insert), "r"(val), "r"(pos), "r"(len));
+
+    ret = (static_cast<unsigned int>(val) << (32 - pos - len)) >> (32 - len);
     return ret;
   }
 };
@@ -498,7 +545,9 @@ struct Bitfield<uint64_t> {
                                                          int pos,
                                                          int len) {
     uint64_t ret;
-    asm("bfe.u64 %0, %1, %2, %3;" : "=l"(ret) : "l"(val), "r"(pos), "r"(len));
+
+
+    ret = (static_cast<uint64_t>(val) << (64 - pos - len)) >> (64 - len);
     return ret;
   }
 
@@ -507,9 +556,9 @@ struct Bitfield<uint64_t> {
                                                          int pos,
                                                          int len) {
     uint64_t ret;
-    asm("bfi.b64 %0, %1, %2, %3, %4;"
-        : "=l"(ret)
-        : "l"(to_insert), "l"(val), "r"(pos), "r"(len));
+
+  ret = (static_cast<uint64_t>(val) << (64 - pos - len)) >> (64 - len);
+
     return ret;
   }
 };
@@ -627,14 +676,20 @@ struct RadixTypeConfig<phi::bfloat16> {
 /*---------------------------Helper Functions------------------*/
 __device__ __forceinline__ int GetLaneId() {
   int lane_id;
-  asm("mov.s32 %0, %%laneid;" : "=r"(lane_id));
-  return lane_id;
+
+// // >>>> PTX2CPP Success <<<<
+// {
+// (lane_id)=(threadIdx.x&(warpSize-1));
+// }
+  return ::__lane_id();
+  // return lane_id;
 }
 
 __device__ __forceinline__ unsigned GetLaneMaskLe() {
   unsigned mask;
-  asm("mov.u32 %0, %%lanemask_le;" : "=r"(mask));
-  return mask;
+
+  return ((uint64_t(1) << ::__lane_id()) << 1) - 1;
+  // return mask;
 }
 
 template <typename T, bool KillDependency, class Function>
@@ -884,7 +939,8 @@ __global__ void GatherKthValue(const T* input,
 
   // 1. Find the k-th value
   T kth_value = static_cast<T>(0);
-  RadixSearch<T, RadixTypeConfig<T>::RadixType, IndexType, false>(
+  // RadixSearch<T, RadixTypeConfig<T>::RadixType, IndexType, false>(
+  RadixSearch<T, typename RadixTypeConfig<T>::RadixType, IndexType, false>(
       cur_input, k, num_cols, shared_mem, &kth_value);
 
   __shared__ int64_t block_min_idx;
@@ -1317,3 +1373,4 @@ bool SortTopk(const phi::GPUContext& dev_ctx,
 }
 }  // namespace funcs
 }  // namespace phi
+//
diff --git a/paddle/phi/kernels/fusion/gpu/fused_layernorm_residual_dropout_bias.h b/paddle/phi/kernels/fusion/gpu/fused_layernorm_residual_dropout_bias.h
index 25c9fa597e..e98d833943 100644
--- a/paddle/phi/kernels/fusion/gpu/fused_layernorm_residual_dropout_bias.h
+++ b/paddle/phi/kernels/fusion/gpu/fused_layernorm_residual_dropout_bias.h
@@ -638,9 +638,6 @@ __global__ __launch_bounds__(THREADS_PER_CTA) void fused_fast_ln_fwd_kernel(
         RandVec<VecSize>(&state, rand);
 #pragma unroll
         for (int jt = 0; jt < VecSize; jt++) {
-#ifndef PADDLE_WITH_HIP
-#pragma unroll
-#endif
           mask_vec[it][jt] = static_cast<MaskType>(rand[jt] >= dropout_prob);
         }
       }
diff --git a/paddle/phi/kernels/fusion/gpu/masked_multihead_attention_kernel.cu b/paddle/phi/kernels/fusion/gpu/masked_multihead_attention_kernel.cu
index e6f66c92f1..b294f7a476 100644
--- a/paddle/phi/kernels/fusion/gpu/masked_multihead_attention_kernel.cu
+++ b/paddle/phi/kernels/fusion/gpu/masked_multihead_attention_kernel.cu
@@ -15,7 +15,7 @@
 #include "paddle/phi/kernels/fusion/gpu/masked_multihead_attention_kernel.h"
 #include "paddle/phi/core/kernel_registry.h"
 #include "paddle/phi/kernels/funcs/aligned_vector.h"
-#include "paddle/phi/kernels/fusion/gpu/mmha_util.cu.h"
+#include "kernels/custom_kernel/mmha_util.cu.h"
 
 namespace phi {
 namespace fusion {
diff --git a/paddle/phi/kernels/fusion/gpu/qkv_unpack_mha_kernel.cu b/paddle/phi/kernels/fusion/gpu/qkv_unpack_mha_kernel.cu
index d947eb1a32..06d0efef4c 100644
--- a/paddle/phi/kernels/fusion/gpu/qkv_unpack_mha_kernel.cu
+++ b/paddle/phi/kernels/fusion/gpu/qkv_unpack_mha_kernel.cu
@@ -15,7 +15,7 @@
 #include "paddle/phi/kernels/fusion/gpu/qkv_unpack_mha_kernel.h"
 #include "paddle/phi/core/kernel_registry.h"
 #include "paddle/phi/kernels/funcs/aligned_vector.h"
-#include "paddle/phi/kernels/fusion/gpu/mmha_util.cu.h"
+#include "kernels/custom_kernel/mmha_util.cu.h"
 
 namespace phi {
 namespace fusion {
diff --git a/paddle/phi/kernels/gpu/gelu_funcs.h b/paddle/phi/kernels/gpu/gelu_funcs.h
index 29fa252e96..4ae72b0935 100644
--- a/paddle/phi/kernels/gpu/gelu_funcs.h
+++ b/paddle/phi/kernels/gpu/gelu_funcs.h
@@ -29,13 +29,13 @@ namespace phi {
 #if defined(__NVCC__) || defined(__HIPCC__)
 template <bool FastMode>
 static __device__ __forceinline__ float FP32FastTanh(float x) {
-#if __CUDA_ARCH__ >= 750 && CUDA_VERSION >= 11000
-  if (FastMode) {
-    float y;
-    asm("tanh.approx.f32 %0,%1; \n\t" : "=f"(y) : "f"(x));
-    return y;
-  }
-#endif
+// #if __CUDA_ARCH__ >= 750 && CUDA_VERSION >= 11000
+//   if (FastMode) {
+//     float y;
+//     asm("tanh.approx.f32 %0,%1; \n\t" : "=f"(y) : "f"(x));
+//     return y;
+//   }
+// #endif
   return tanhf(x);
 }
 
diff --git a/paddle/phi/kernels/gpu/lstsq_kernel.cu b/paddle/phi/kernels/gpu/lstsq_kernel.cu
index 75a8f71d8c..cb21e9e301 100644
--- a/paddle/phi/kernels/gpu/lstsq_kernel.cu
+++ b/paddle/phi/kernels/gpu/lstsq_kernel.cu
@@ -21,7 +21,7 @@
 #include "paddle/phi/core/kernel_registry.h"
 #include "paddle/phi/kernels/full_kernel.h"
 #include "paddle/phi/kernels/funcs/slice.h"
-#include "paddle/phi/kernels/impl/lstsq_kernel_impl.h"
+#include "kernels/impl/lstsq_kernel_impl.h"
 #include "paddle/phi/kernels/impl/qr_kernel_impl.h"
 #include "paddle/phi/kernels/impl/tril_triu_kernel_impl.h"
 #include "paddle/phi/kernels/lstsq_kernel.h"
diff --git a/paddle/phi/kernels/impl/gammaincc_kernel_impl.h b/paddle/phi/kernels/impl/gammaincc_kernel_impl.h
index 4a28600c38..d96495b7aa 100644
--- a/paddle/phi/kernels/impl/gammaincc_kernel_impl.h
+++ b/paddle/phi/kernels/impl/gammaincc_kernel_impl.h
@@ -56,8 +56,8 @@ HOSTDEVICE T igam(const T a, const T x) {
 
 template <typename T>
 HOSTDEVICE T igamc(const T a, const T x) {
-  static T big = 4.503599627370496e15;
-  static T biginv = 2.22044604925031308085e-16;
+  const static T big = 4.503599627370496e15;
+  const static T biginv = 2.22044604925031308085e-16;
 
   if ((x <= T{0}) || (a <= T{0})) return (T{1.0});
 
diff --git a/paddle/phi/kernels/impl/gammaln_grad_kernel_impl.h b/paddle/phi/kernels/impl/gammaln_grad_kernel_impl.h
index c627cc1264..b3941570ee 100644
--- a/paddle/phi/kernels/impl/gammaln_grad_kernel_impl.h
+++ b/paddle/phi/kernels/impl/gammaln_grad_kernel_impl.h
@@ -20,8 +20,8 @@
 namespace phi {
 template <typename T>
 HOSTDEVICE T digamma_positive_domain(T x) {
-  static T c = T{8.5};
-  static T euler_mascheroni = T{0.57721566490153286060};
+  const static T c = T{8.5};
+  const static T euler_mascheroni = T{0.57721566490153286060};
   T r;
   T value;
   T x2;
@@ -54,7 +54,7 @@ HOSTDEVICE T digamma_positive_domain(T x) {
 
 template <typename T>
 HOSTDEVICE T digamma(T x) {
-  static T pi = T{3.14159265358979323846};
+  const static T pi = T{3.14159265358979323846};
 
   if (x == T{0.0}) {
     T inf = std::numeric_limits<T>::infinity();

diff --git a/paddle/phi/kernels/gpudnn/softmax_gpudnn.h b/paddle/phi/kernels/gpudnn/softmax_gpudnn.h
index be6ee4f854..1f507c99f4 100644
--- a/paddle/phi/kernels/gpudnn/softmax_gpudnn.h
+++ b/paddle/phi/kernels/gpudnn/softmax_gpudnn.h
@@ -111,7 +111,7 @@ __device__ __forceinline__ void WarpReduceSum(T* sum) {
 #pragma unroll
     for (int i = 0; i < BatchSize; ++i) {
       T sum_val =
-          phi::backends::gpu::CudaShuffleXorSync(0xFFFFFFFF, sum[i], offset);
+          phi::backends::gpu::CudaShuffleXorSync(0xFFFFFFFFFFFFFFFFULL, sum[i], offset);
       sum[i] = sum[i] + sum_val;
     }
   }
@@ -124,7 +124,7 @@ __device__ __forceinline__ void WarpReduceMax(T* sum) {
 #pragma unroll
     for (int i = 0; i < BatchSize; ++i) {
       T max_val =
-          phi::backends::gpu::CudaShuffleXorSync(0xFFFFFFFF, sum[i], offset);
+          phi::backends::gpu::CudaShuffleXorSync(0xFFFFFFFFFFFFFFFFULL, sum[i], offset);
       sum[i] = max(sum[i], max_val);
     }
   }
@@ -2678,27 +2678,7 @@ void SoftmaxForwardCUDAKernelDriverImpl(const GPUContext& dev_ctx,
       using T4 = typename VecT4<T>::Type;
       using T2 = typename VecT2<T>::Type;
 
-      if (dim % 4 == 0) {
-        SwitchWarpSoftmaxForward<T, T4, IndexType, LogMode>(blocks,
-                                                            threads,
-                                                            dev_ctx,
-                                                            out_data,
-                                                            x.data<T>(),
-                                                            N,
-                                                            dim,
-                                                            dim,
-                                                            dim_log2);
-      } else if (dim % 2 == 0) {
-        SwitchWarpSoftmaxForward<T, T2, IndexType, LogMode>(blocks,
-                                                            threads,
-                                                            dev_ctx,
-                                                            out_data,
-                                                            x.data<T>(),
-                                                            N,
-                                                            dim,
-                                                            dim,
-                                                            dim_log2);
-      } else {
+      if (std::is_same<T, float>::value) {
         SwitchWarpSoftmaxForward<T, T, IndexType, LogMode>(blocks,
                                                            threads,
                                                            dev_ctx,
@@ -2708,14 +2688,41 @@ void SoftmaxForwardCUDAKernelDriverImpl(const GPUContext& dev_ctx,
                                                            dim,
                                                            dim,
                                                            dim_log2);
-      }
-    } else {
-      if (dim >= MATRIX_SOFTMAX_THRESHOLD) {
-        LaunchKeMatrixSoftmaxForwardKernel<T, IndexType, LogMode>(
-            dev_ctx, out_data, x.data<T>(), N, dim);
       } else {
-        LaunchSoftmaxForwardCudnnKernel<T>(dev_ctx, x, axis, LogMode, out);
+        if (dim % 4 == 0) {
+          SwitchWarpSoftmaxForward<T, T4, IndexType, LogMode>(blocks,
+                                                              threads,
+                                                              dev_ctx,
+                                                              out_data,
+                                                              x.data<T>(),
+                                                              N,
+                                                              dim,
+                                                              dim,
+                                                              dim_log2);
+        } else if (dim % 2 == 0) {
+          SwitchWarpSoftmaxForward<T, T2, IndexType, LogMode>(blocks,
+                                                              threads,
+                                                              dev_ctx,
+                                                              out_data,
+                                                              x.data<T>(),
+                                                              N,
+                                                              dim,
+                                                              dim,
+                                                              dim_log2);
+        } else {
+          SwitchWarpSoftmaxForward<T, T, IndexType, LogMode>(blocks,
+                                                             threads,
+                                                             dev_ctx,
+                                                             out_data,
+                                                             x.data<T>(),
+                                                             N,
+                                                             dim,
+                                                             dim,
+                                                             dim_log2);
+        }
       }
+    } else {
+      LaunchSoftmaxForwardCudnnKernel<T>(dev_ctx, x, axis, LogMode, out);
     }
   } else {
     LaunchNormalSoftmaxForward<T, IndexType, LogMode>(

diff --git a/paddle/phi/kernels/gpu/cross_entropy_grad_kernel.cu b/paddle/phi/kernels/gpu/cross_entropy_grad_kernel.cu
index 0a415200df..b0732e28f3 100644
--- a/paddle/phi/kernels/gpu/cross_entropy_grad_kernel.cu
+++ b/paddle/phi/kernels/gpu/cross_entropy_grad_kernel.cu
@@ -147,7 +147,7 @@ void CrossEntropyWithSoftmaxGradGPUKernel(const GPUContext& dev_ctx,
                                           DenseTensor* logits_grad) {
   PADDLE_ENFORCE_EQ(
       dev_ctx.GetPlace().GetType(),
-      AllocationType::GPU,
+      AllocationType::CUSTOM,
       common::errors::Unavailable("softmax_with_cross_entropy operator's "
                                   "CUDA kernel only runs on GPU device."));
   const T* loss_grad_data = loss_grad.data<T>();
diff --git a/paddle/phi/kernels/funcs/cublaslt.h b/paddle/phi/kernels/funcs/cublaslt.h
index d8bc15926b..6071baf340 100644
--- a/paddle/phi/kernels/funcs/cublaslt.h
+++ b/paddle/phi/kernels/funcs/cublaslt.h
@@ -40,19 +40,19 @@ class CublasLtHelper {
   CublasLtHelper(int m, int k, int n, cublasLtHandle_t handle)
       : handle_(handle), alpha_(1), beta_(0), m_(m), k_(k), n_(n) {
     cublasStatus_t status;
-#if CUBLAS_VER_MAJOR < 11
-    cudaDataType_t cudaComputeType = CUDA_R_32I;
-#else
+// #if CUBLAS_VER_MAJOR < 11
+//     cudaDataType_t cudaComputeType = CUDA_R_32I;
+// #else
     cublasComputeType_t cudaComputeType = CUBLAS_COMPUTE_32I;
-#endif
+// #endif

     // matmul desc
-#if CUBLAS_VER_MAJOR < 11
-    status = dynload::cublasLtMatmulDescCreate(&matmul_desc_, cudaComputeType);
-#else
+// #if CUBLAS_VER_MAJOR < 11
+//     status = dynload::cublasLtMatmulDescCreate(&matmul_desc_, cudaComputeType);
+// #else
     status = dynload::cublasLtMatmulDescCreate(
         &matmul_desc_, cudaComputeType, CUDA_R_32I);
-#endif
+// #endif

     PADDLE_ENFORCE_EQ(
         status,
         
