diff --git a/paddle/common/backend_header.h b/paddle/common/backend_header.h
index 82f5aca792..b16cb80b20 100644
--- a/paddle/common/backend_header.h
+++ b/paddle/common/backend_header.h
@@ -23,6 +23,13 @@
 #include <cuda_bf16.h>
 #endif
 
+#if defined(__MUSACC__)
+#define PADDLE_MUSA_BF16
+#include <musa.h>
+#include <musa_bf16.h>
+#endif       
+
+
 #ifndef PADDLE_WITH_HIP
 #if !defined(_WIN32)
 #define PADDLE_ALIGN(x) __attribute__((aligned(x)))
